---
title: "[Last 2000909, Final 191205-1212, 18project, 3T3] BRBseq0432lane2_QC_tmpl_v6_noumi_H3mm18_Dox_linear_0718_fin191205_last200811_add200910 (umi補正なし,fdr0.1) (TPM,尤度比検定)"
output:
  html_notebook: 
    toc: yes
  pdf_document: 
    keep_tex: yes
    latex_engine: lualatex
---

### Setup

```{r libraries,message=FALSE}
library(ggplot2)
library(dplyr)
library(tidyr)
library(purrr)
source("/home/guestA/n70275b/work/rscripts/geomNorm.R")

# Helper function
#ggpoints <- function(x,...) 
#  ggplot(x,...) + geom_point(size=3,stroke=1) +
#  ggrepel::geom_text_repel(size=4) + theme_minimal() + mycolor

## ラベルあり
ggpoints <- function(x,...) 
  ggplot(x,...) + geom_point(stroke=1) +
  ggrepel::geom_text_repel(size=4) + theme_minimal() + mycolor

## ラベルなし
#ggpoints <- function(x,...) 
#  ggplot(x,...) + geom_point(stroke=1) + theme_minimal() + mycolor

print(sessionInfo(),locale=FALSE)

select <- dplyr::select
count <- dplyr::count
rename <- dplyr::rename
```

### Parameters

*modify here*

```{r params}
# Files


#deftable <- "~/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/deftable_BRB_noumi_new_190520_fin191205ver.txt" #Umi補正なし (BRB)

deftable <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/deftable_BRB_noumi_new_190520_Last20200811ver.txt"

#deftable <- "~/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/deftable_BRB_noumi_new_190520_fin191205ver.txt" #Umi補正なし (BRB)


#deftable <- "deftable_BRB_noumi_new_190520.txt" #Umi補正なし (BRB)

#deftable <- "~/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/deftable_BRB_noumi_new_190520.txt" #Umi補正なし (BRB)


## Data selection (filter rows of deftable)
#use <- quo(!grepl("^18",group) & (group != "Nc-minusTryd"))
#use <- quo(TRUE) # use all
use <- quo(type != "C2C12")

# Species specific parameters
species <- "Mus musculus"
biomartann <- "mmusculus_gene_ensembl"
maxchrom <- 19 # 19: mouse, 22: human


# Graphics
# aesthetic mapping of labels
#myaes <- aes(colour=enzyme,shape=leg,label=rep) 
#myaes <- aes(colour=growth,shape=type,size=count) #ラベルなし
#myaes <- aes(colour=growth,shape=type,label=replicate,size=count) #ラベルあり
#myaes <- aes(colour=enzyme,shape=leg,label=replicate) #ラベルあり
#myaes <- aes(colour=enzyme,shape=leg,label=factor(rep))
#myaes <- aes(colour=type, shape=revcro, label=read, size=count)
#myaes <- aes(colour=type, shape=revcro, label=read)

#myaes <- aes(colour=growth,shape=type,label=replicate,size=count) #ラベルあり
#myaes <- aes(colour=time,shape=type,size=count,label=replicate)
#myaes <- aes(colour=WT_KO_intact_CTX, shape=Day,size=count,label=f_m)

#myaes <- aes(colour=WT_KO_intact_CTX, shape=Day, label=f_m) #サイズを変えず
#myaes <- aes(colour=growth,shape=type,label=replicate,size=count) #ラベルあり
myaes <- aes(colour=time,shape=type,label=rep,size=count) #ラベルあり
myaes2 <- aes(colour=time,shape=type) #kuwa add

# color palette of points: See vignette("ggsci")
mycolor <- ggsci::scale_color_aaas()



# PCA/UMAP
scalerows <- TRUE # gene-wise scaling (pattern is the matter?)
ntop <- 500 # number of top-n genes with high variance
seed <- 123 # set another number if UMAP looks not good
n_nei <- 6  # number of neighboring data points in UMAP #ここをどうしたらいい？


# DESeq2
#model <- ~groupn+lead #dateも追加
#model <- ~leg + enzyme + leg:enzyme
#model <- ~type+growth#+type:growth
#model <- ~group+lead


#model <- ~group
#model <- ~type+growth+type:growth #これでは相互作用が入っていない
#model <- ~type+growth #これでは相互作用が入っていない


model <- ~group
#model <- ~type+growth+growth:type

fdr <- 0.1 # acceptable false discovery rate
lfcthreth <- log2(1) # threshold in abs(log2FC)

# controls should be placed in the right
contrast <- list(
  
  group_UI_Doxplus_vs_minus = c("group", "BRB_UI_DoxPlus", "BRB_UI_DoxMinus"),
  group_0h_Doxplus_vs_minus = c("group", "BRB_0h_DoxPlus", "BRB_0h_DoxMinus"),
  group_24h_Doxplus_vs_minus = c("group", "BRB_24h_DoxPlus", "BRB_24h_DoxMinus"),
  group_48h_Doxplus_vs_minus = c("group", "BRB_48h_DoxPlus", "BRB_48h_DoxMinus")
  
  
  #group_UI_Doxplus_vs_minus = c("group", "Doxplus_UI", "Doxminus_UI"),
  #group_Diff0h_Doxplus_vs_minus = c("group", "Doxplus_Diff0h", "Doxminus_Diff0h"),
  #group_Diff24h_Doxplus_vs_minus = c("group", "Doxplus_Diff24h", "Doxminus_Diff24h"),
  #group_Diff48h_Doxplus_vs_minus = c("group", "Doxplus_Diff48h", "Doxminus_Diff48h")
  
  
  #Intercept = list("Intercept"), # reference level
  #leg_LvsR = c("leg", "L", "R"),
  #enz_KvsC = c("enzyme","K","C")
  #legL.enzK = list("legL.enzymeK") # interaction
  
  #type_Doxplus_vs_minus = c("type", "Doxplus", "Doxminus")
)
```



### Retrieve Biomart

```{r biomart, cache=TRUE}
#if(!exists("e2g")){
#  #ensembl <- biomaRt::useMart("ENSEMBL_MART_ENSEMBL",host="asia.ensembl.org")
#  #ensembl <- biomaRt::useMart("ENSEMBL_MART_ENSEMBL",host="uswest.ensembl.org")
#  ensembl <- biomaRt::useMart("ENSEMBL_MART_ENSEMBL",host="useast.ensembl.org")
#  mart <- biomaRt::useDataset(biomartann,mart=ensembl)
#  e2g <- biomaRt::getBM(attributes=c("ensembl_gene_id","external_gene_name",
#    "gene_biotype","chromosome_name"), mart=mart) %>% as_tibble %>%
#  rename(
#    ens_gene = ensembl_gene_id,
#    ext_gene = external_gene_name,
#    biotype = gene_biotype,
#    chr = chromosome_name
#  )
#}

e2g <- readr::read_csv("/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/ensemble_list_useast.csv")

annotate <- partial(right_join,e2g,by="ens_gene")

#annotate <- partial(right_join,e2g,by="ens_gene")

#-----#
nrow(e2g)
#readr::write_csv(e2g,"ensemble_list_asia.csv")
#readr::write_csv(e2g,"ensemble_list_uswest.csv")
#readr::write_csv(e2g,"ensemble_list_useast.csv")
```

### Load counts

```{r loadUMI}
def <- readr::read_tsv(deftable) %>% filter(!!use)
print(def)

#def$growth <- factor(def$growth,levels =c("UI","Diff0h","Diff24h","Diff48h"))
#def$type <- factor(def$type,levels =c("Doxminus","Doxplus"))

#factor(def$growth,levels =c("UI","Diff0h","Diff24h","Diff48h"))
# [1] UI      UI      UI      UI      UI      UI      UI      UI      Diff0h  Diff0h  Diff0h  Diff0h  Diff0h  Diff0h  Diff0h  Diff0h  Diff24h Diff24h Diff24h Diff24h
#[21] Diff24h Diff24h Diff24h Diff24h Diff48h Diff48h Diff48h Diff48h Diff48h Diff48h Diff48h Diff48h
#Levels: UI Diff0h Diff24h Diff48h


####--- New ---#### (no UMI ?)
# Set reference levels according to the contrast
for(x in keep(contrast,is.character))
  def[[x[1]]] <- relevel(factor(def[[x[1]]]),x[3])

umi <- def$file %>% unique %>% tibble(file=.) %>% 
  dplyr::mutate(data=map(file,readr::read_tsv,progress=FALSE)) %>%
  unnest() %>% dplyr::rename(barcode=cell) %>%
  dplyr::inner_join(select(def,file,barcode,sample),.,c("file","barcode")) %>%
  select(-file,-barcode) %>% dplyr::rename(ens_gene=gene)

print(umi)

## sample, barcode, file を忘れずに！

mat <- umi %>% annotate %>%
  dplyr::mutate(chr=factor(chr,c(1:maxchrom,"X","Y","MT"))) %>%
  filter(!is.na(chr)) %>% spread(sample,count,fill=0)

## to check read vias, this add read number as "n" column (2019/4/19)
def <- umi %>% count(sample,wt=count) %>% dplyr::inner_join(def,.) %>% dplyr::rename(count=n)
####-----------#### 




# Set reference levels according to the contrast
#for(x in keep(contrast,is.character))
#  def[[x[1]]] <- relevel(factor(def[[x[1]]]),x[3])

#umi <- def$file %>% unique %>% tibble(file=.) %>% 
#  mutate(data=map(file,readr::read_tsv,progress=FALSE)) %>%
#  unnest() %>% dplyr::rename(barcode=cell) %>%
#  inner_join(select(def,file,barcode,sample),.,c("file","barcode")) %>%
#  select(-file,-barcode) %>% dplyr::rename(ens_gene=gene)

#mat <- umi %>% annotate %>%
#  mutate(chr=factor(chr,c(1:maxchrom,"X","Y","MT"))) %>%
#  filter(!is.na(chr)) %>% spread(sample,count,fill=0)

print(mat)

## to check read vias, this add read number as "n" column (2019/4/19)
#def <- umi %>% count(sample,wt=count) %>% inner_join(def,.) %>% dplyr::rename(count=n)

print(def)


##====================================##
# 確認 (20191204) ２つの値は一緒か？
# 生のデータカウント中の遺伝子総数

umi %>% group_by(ens_gene) %>% summarise %>% nrow()

umi %>% spread(sample,count,fill=0) %>% nrow()

mat %>% nrow()
mat %>% filter(chr!="MT") %>% nrow() # MTなし

# matでは、chr等が不明なものは省いている。
# DEGでは、さらにMTも省いている。
##====================================##

```

### Reads breakdown

#### Total reads

```{r totalReads, fig.width=7,fig.height=5}
bychr <- mat %>% select(-(1:3)) %>%
  gather("sample","count",-chr) %>%
  group_by(chr,sample) %>% summarise(total=sum(count)) %>% ungroup

ggplot(bychr,aes(reorder(sample,dplyr::desc(sample)),total/1e6,fill=chr)) +
  theme_linedraw() + geom_bar(stat="identity") + coord_flip() +
  xlab("sample") + ylab("million reads") + ggsci::scale_fill_igv() +
  scale_x_discrete(limits = rev(levels(sample)))


```

#### Biotype

```{r biotype,fig.width=8,fig.height=7}
bt <- mat %>% select(-c(1,2,4)) %>% group_by(biotype) %>%
  summarise_all(sum) %>% filter_at(-1,any_vars(. > 1000))
bt %>% tibble::column_to_rownames("biotype") %>%
  as.matrix %>% t %>% mosaicplot(las=2,shade=TRUE)
```

### Correlations

drop rows with all 0 -> +1/2 -> geom.scale -> log -> Pearson's

```{r makemat, fig.width=8,fig.height=7}
matf <- mat %>% filter(chr!="MT") %>% filter_at(-(1:4),any_vars(. > 0))
X <- matf %>% select(-(1:4)) %>% as.matrix
rownames(X) <- matf$ens_gene
lX <- log(gscale(X+0.5))
R <- cor(lX); diag(R) <- NA
pheatmap::pheatmap(R,color=viridis::viridis(256))
```

### Dimension reduction

```{r PCA,fig.width=4,fig.height=3}
# set scale=TRUE if the patterns (not level) is the matter
p <- prcomp(t(lX[rank(-apply(lX,1,var)) <= ntop,]),scale=scalerows,center=TRUE)
screeplot(p,las=2,main="Importance")
print(summary(p)$imp[,seq(min(10,ncol(X)))])
```

```{r makescoreDF}
label <- def %>% filter(sample %in% colnames(X))
df <- data.frame(p$x) %>% as_tibble(rownames="sample") %>%
  inner_join(label,.) %>% select(-file)

print(df)
```

```{r proximity,fig.width=6,fig.height=4}
ggpoints(df,modifyList(aes(PC1,PC2),myaes))
set.seed(seed)
um <- uwot::umap(p$x,n_nei,2)
df <- as_tibble(um) %>% rename(UMAP1=V1,UMAP2=V2) %>% bind_cols(df)
ggpoints(df,modifyList(aes(UMAP1,UMAP2),myaes))

print(df)

##  kuwakado 変更 ##
ggpoints <- function(x,...) 
  ggplot(x,...) + geom_point(stroke=1) + theme_minimal() + mycolor

#ggpoints(df,modifyList(aes(PC1,PC2),myaes2))
#set.seed(seed)
#um <- uwot::umap(p$x,n_nei,2)
#df <- as_tibble(um) %>% rename(UMAP1=V1,UMAP2=V2) %>% bind_cols(df)
#ggpoints(df,modifyList(aes(UMAP1,UMAP2),myaes2))
## ## ## ##
```

### DESeq2

#### Fit model

```{r deseq2}
dds <- DESeq2::DESeqDataSetFromMatrix(X[,label$sample],label,model)
dds <- DESeq2::DESeq(dds)


#=====#

dds <- DESeq2::estimateSizeFactors(dds)
norm <- DESeq2::counts(dds,normalized=TRUE) #DEGを取った後のクラスタリングに使う。

normalizedcount <- as.data.frame(norm) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble
readr::write_csv(normalizedcount, "./H3mm18KO_3T3_Dox_normCount.csv")

normalizedcount %>% inner_join(e2g, by = "ens_gene")  %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(label$sample)) %>% readr::write_csv("./H3mm18KO_3T3_Dox_normCount_genename.csv")

#count_dds <- estimateSizeFactors(dds)
#counts(count_dds, normalized=TRUE)

####--- + size factors を書き出し ------------------####
as.data.frame(DESeq2::sizeFactors(dds))  %>% tibble::rownames_to_column("sample") %>% readr::write_csv("./H3mm18KO_3T3_Dox__sizefactors.csv")
sizefactors_sample <- as.data.frame(DESeq2::sizeFactors(dds))  %>% tibble::rownames_to_column("sample")

sizefactors_sample

```


vst => z score

```{r zscore 200811add}

vsd <- DESeq2::vst(dds) #normalized countが入っている。(vstかrlog)
Xd <- SummarizedExperiment::assay(vsd) # 全て選択(200326) 20190920を元に (191024)
Xs <- Xd %>% t %>% scale %>% t

vst_score <- Xd %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble #200909 add
vst_type <- vst_score  %>% annotate %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(label$sample))


zscore <- Xs %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble
zscore_type <- zscore  %>% annotate %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(label$sample))

readr::write_csv(vst_score, "H3mm18KO_3T3_Dox__vst_all.csv") #200909 add
readr::write_csv(vst_type, "H3mm18KO_3T3_Dox__vst_type_all.csv") #200909 add
readr::write_csv(zscore, "H3mm18KO_3T3_Dox__zscore_all.csv")
readr::write_csv(zscore_type, "H3mm18KO_3T3_Dox__zscore_type_all.csv")

nrow(vst_type)
nrow(zscore_type)

```

#### Diagnostics plot

```{r diagnostics,fig.width=7,fig.height=5}
DESeq2::sizeFactors(dds) %>%
  {tibble(sample=names(.),sizeFactor=.)} %>%
  ggplot(aes(sample,sizeFactor)) + theme_minimal() +
  geom_bar(stat="identity") + coord_flip()
DESeq2::plotDispEsts(dds)
```

--------------------
ここから 20200909 add

### Dimension reduction (VST)

```{r PCA vst,fig.width=4,fig.height=3}
# set scale=TRUE if the patterns (not level) is the matter
## Xd: vst values

p_vst <- prcomp(t(Xd[rank(-apply(Xd,1,var)) <= ntop,]),scale=scalerows,center=TRUE) ## ここでは、zscoreを使ってPCAを行なっているのと同じ。
screeplot(p_vst,las=2,main="Importance")
print(summary(p_vst)$imp[,seq(min(10,ncol(X)))])

## save data

var_vst_all <- apply(Xd,1,var) %>% as.data.frame()  %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::rename(var=".") 
var_vst_all %>% readr::write_csv("./PCA/var_vst_all.csv")
var_vst_all %>% nrow()

rankvar_vst_all <- rank(-apply(Xd,1,var)) %>% as.data.frame()  %>% tibble::rownames_to_column("ens_gene") %>% as_tibble  %>%  dplyr::rename(ranking_var=".") 
rankvar_vst_all %>% readr::write_csv("./PCA/rankvar_vst_all.csv")
rankvar_vst_all %>% nrow()

vst__rankvartop500gene <- Xd[rank(-apply(Xd,1,var)) <= ntop,] %>% as.data.frame()  %>% tibble::rownames_to_column("ens_gene") %>% as_tibble 
vst__rankvartop500gene %>% readr::write_csv("./PCA/H3mm18KO_3T3_Dox__vst__rankvartop500gene.csv")
vst__rankvartop500gene %>% nrow()


```
```{r save pca data}

p_vst_scale <- p_vst$scale %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble 
p_vst_sdev <- p_vst$sdev %>% as.data.frame() %>% tibble::rownames_to_column("PC") %>% as_tibble  %>%  dplyr::rename(Standard_deviation=".") 
p_vst_x <- p_vst$x %>% as.data.frame()  %>% tibble::rownames_to_column("sample") %>% as_tibble 
p_vst_center <- p_vst$center %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble 
p_vst_rotation <- p_vst$rotation %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble 

p_vst_scale %>% readr::write_csv("./PCA/H3mm18KO_3T3_Dox__vartop500gene__pca_vst_scale.csv")
p_vst_sdev %>% readr::write_csv("./PCA/H3mm18KO_3T3_Dox__vartop500gene__pca_vst_sdev.csv")
p_vst_x %>% readr::write_csv("./PCA/H3mm18KO_3T3_Dox__vartop500gene__vst__pca_vst_x.csv")
p_vst_center %>% readr::write_csv("./PCA/H3mm18KO_3T3_Dox__vartop500gene__vst__pca_vst_center.csv")
p_vst_rotation %>% readr::write_csv("./PCA/H3mm18KO_3T3_Dox__vartop500gene__vst__pca_vst_rotation.csv")

```



```{r makescoreDF vst}
#label <- def %>% filter(sample %in% colnames(X))
df_vst <- data.frame(p_vst$x) %>% as_tibble(rownames="sample") %>%
  inner_join(label,.) %>% select(-file)

df_vst <- df_vst  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% mutate(type=factor(type,c("DoxPlus","DoxMinus"))) %>% mutate(rep=factor(rep, c("1", "2", "3", "4"))) %>% mutate(group=factor(group, c("BRB_UI_DoxMinus", "BRB_UI_DoxPlus", "BRB_0h_DoxMinus","BRB_0h_DoxPlus","BRB_24h_DoxMinus","BRB_24h_DoxPlus","BRB_48h_DoxMinus","BRB_48h_DoxPlus"))) # 20200908
print(df_vst)

df_vst %>% readr::write_csv("./PCA/PCA_vst_vartop500gene.csv")
ggpoints(df_vst,modifyList(aes(PC1,PC2),myaes))

```



```{r color pal PCA, fig.width=5,fig.height=4}
cols_pca <- brewer.pal(8, "Paired")

mmplot <- df_vst %>% ggplot(aes(x=PC1,y=PC2,color=group)) + geom_abline(intercept=0,slope=0,colour="gray") + geom_vline(xintercept=0,colour="gray") + geom_point(stroke=1) + theme_bw() +  scale_color_manual(values = cols_pca) + ggtitle("var top 500 genes")  + theme(legend.position = "top",legend.text = element_text(size=3),legend.title = element_blank()) + coord_fixed(ratio=1)

mmplot
ggsave(plot = mmplot,file="./PCA/PC1PC2_vst_vartop500gene.pdf", dpi = 100, width = 4, height = 5)


mmplot <- df_vst %>% ggplot(aes(x=PC1,y=type,color=group)) + geom_vline(xintercept=0,colour="gray") + geom_point(stroke=1) + theme_bw() +  scale_color_manual(values = cols_pca) + ggtitle("var top 500 genes")   + theme(legend.position = "top",legend.text = element_text(size=3),legend.title = element_blank())

mmplot
ggsave(plot = mmplot,file="./PCA/PC1_vst_vartop500gene.pdf", dpi = 100, width = 4, height = 2)


```


### Check (vs 20200811 data)

(vstの値は200811の時に出力していなかったので、sizefactorとzsocreが同じになってるかどうかで判定)

```{r comp 200811 data}
sizefactors_sample_old <- readr::read_csv("/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/H3mm18KO_3T3_Dox__sizefactors.csv")
zscore_old <- readr::read_csv("/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/H3mm18KO_3T3_Dox__zscore_all.csv")

```
PC1_groupavg <- df_vst %>% group_by(group) %>% summarise(PC1_avg = mean(PC1),PC1_sd=sd(PC1),PC1_V=sd(PC1)*sd(PC1), PC1_S=sd(PC1)*sd(PC1)*(4-1))

#PC1_group2 <- PC1_group %>% dplyr::select(sample, group,time,type,seq,rep,PC1,PC1_avg,PC1_sd)  %>% group_by(group,sample,time,type) %>% summarise(PC1,PC1_avg,PC1_sd, #PC1_d=PC1-PC1_avg,PC1_d2=(PC1-PC1_avg)*(PC1-PC1_avg)) %>% ungroup()
#PC1_group2 %>% group_by(group,time,type) %>% summarise(PC1_S=sum(PC1_d2)) %>% ungroup()

```{r ddddd}

PC1_Sw <- df_vst %>% group_by(group,time,type) %>% summarise(PC1_avg = mean(PC1),PC1_sd=sd(PC1),PC1_V=sd(PC1)*sd(PC1), PC1_Sk2=sd(PC1)*sd(PC1)*(4-1)) %>% ungroup()
#PC1_Sw %>% dplyr::select(group,PC1_avg) %>% spread(key=group,value=PC1_avg)
#PC1_Sw_UI <- PC1_Sw %>% filter(time=="UI") 

Time_list <- c("UI","0h","24h","48h")


#######
print("~~ H3p3_BRB ~~")

for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clusterAll: H3p3_BRB --"))
  sb1 <- PC1_Sw %>% filter((time==Time_list[i]))
  sb2 <- (sb1$PC1_avg[1]-sb1$PC1_avg[2])^2
  sw1 <- sum(sb1$PC1_Sk2[1]+sb1$PC1_Sk2[2])
  print(sb2)
  
  if (i == 1) { 
    sb_r <- data.frame(time=Time_list[i],group1=sb1$group[1],group2=sb1$group[2],type1=sb1$type[1],type2=sb1$type[2],PC1_Sb=sb2,PC1_Sw=sw1)%>% as_tibble()
  } 
  else {
    sb_3 <- data.frame(time=Time_list[i],group1=sb1$group[1],group2=sb1$group[2],type1=sb1$type[1],type2=sb1$type[2],PC1_Sb=sb2,PC1_Sw=sw1)%>% as_tibble()
    sb_r <- bind_rows(sb_r, sb_3)
  }
}

PC1_Sb <- sb_r

print(PC1_Sb)

PC1_SwSb <- left_join(PC1_Sw,PC1_Sb) %>% mutate(PC1_Jw=PC1_Sb/PC1_Sw)

print(PC1_SwSb)

```

```{r PC1 plot}

mmplot <- df_vst %>% ggplot(aes(x=PC1,y=type,color=group)) + geom_vline(xintercept=0,colour="gray")+ theme_bw() +  scale_color_manual(values = cols_pca) + ggtitle("var top 500 genes")   + theme(legend.position = "top",legend.text = element_text(size=3),legend.title = element_blank())+ geom_point(aes(x=PC1_avg,y=type),data=PC1_SwSb)

mmplot
ggsave(plot = mmplot,file="./PCA/PC1_vst_vartop500gene__with_Jw.pdf", dpi = 100, width = 4, height = 2)


```

