---
title: "TSSpm5kb__3T3_EGFP18_Dox_20200811__mainH3p3_ChIL01100111_ATAC0049_BRB0432L2"
output: html_notebook
author: "Kuwakado"
date: "2020/8/11"
---


2020.8.11 BRBを再解析(log2FCを計算するため)、それに合わせて再解析

```{r setting}

print(Sys.Date())
print(sessionInfo(),locale=FALSE)

```

```{r setup 0}
#{r setup 0, include=FALSE} knitr::opts_chunk$set(echo = TRUE)

library(ggplot2)
library(dplyr)
library(tidyr)
library(purrr)
source("/home/guestA/n70275b/work/rscripts/geomNorm.R")

## ラベルあり
ggpoints <- function(x,...) 
  ggplot(x,...) + geom_point(stroke=1) +
  ggrepel::geom_text_repel(size=4) + theme_minimal() + mycolor

maxchrom <- 19 # 19: mouse, 22: human

mycolor <- ggsci::scale_color_aaas()

# PCA/UMAP
scalerows <- TRUE # gene-wise scaling (pattern is the matter?)
ntop <- 500 # number of top-n genes with high variance
seed <- 123 # set another number if UMAP looks not good
n_nei <- 6  # number of neighboring data points in UMAP #ここをどうしたらいい？


#----------------#

#cluster_num <- 4 

filepath_summary <- "/home/guestA/o70578a/akuwakado/kuwakado/ChILSeq2/Komatsu_3T3_EGFP_H3mm18_Dox_chIl_0111NOVAseq/TSS_count/ChILAll_TSS_pm5kb_withATAC/ChIL01100111_ATAC0049L1__3T3_EGFP18_Dox__TSS_pm5kb_20200624.count.txt"

# deftable 修正版
filepath_def <- "/home/guestA/o70578a/akuwakado/kuwakado/ChILSeq2/Komatsu_3T3_EGFP_H3mm18_Dox_chIl_0111NOVAseq/TSS_count/ChILAll_TSS_pm5kb_withATAC/deftable_multicov_ChIL01100111_20200501_3T3_EGFP18_UI_DoxMinus_H3p3K27acK4Kme327me3_withATAC.txt"



#--- サンプルの選択 ---#

folder_name <- "ChIL H3.3, H3K27ac, H3K4me3, H3K27me3, ATAC" #"ChIL_H3K27me3_H3K27me3"
#remove_sample=c("Doxminus_UI_ATAC_4","Doxplus_UI_ATAC_4","Doxminus_D48_ATAC_1","Doxminus_D48_ATAC_4","Doxplus_D48_ATAC_4") #このサンプルを削除(20190917) <ATACの場合>

#use <- quo(!(sample %in% remove_sample)) #このサンプルを削除(20190917)

#--- multiBamSummary の略 ---#
# データ保存用のpath
#csvfilepath <- basename(filepath_summary) %>% sub(".count.txt", "__", .)
csvfilepath <- basename(filepath_summary) %>% sub(".count.txt", "", .)  %>% sub("ChIL01100111_ATAC0049L1__3T3_EGFP18_Dox__", "", .)
print(csvfilepath)


```

### ensembleのデータの読み込み

```{r biomart}
filepath_BRBensemble <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/ensemble_list_useast.csv" #BRBの時のgene名リスト

ensemble <- readr::read_csv(filepath_BRBensemble) %>% mutate_if(is.double, as.integer)

annotate <- partial(right_join,ensemble,by="ens_gene") #2gunで使う

```



--------

### Select regions


UCSCの形式の場合 (20191016)

#### unite tables

```{r setup mulcov 1}

# 20200617

def_list <- readr::read_tsv(filepath_def) %>% mutate(seq=factor(seq, c("ATAC","H3p3", "H3K27ac","H3K4me3","H3K27me3")))  %>% 
mutate(time1 = time, rep1=rep) %>% unite(time1,rep1,col="time_replicate")  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% mutate(type=factor(type,c("DoxPlus","DoxMinus"))) %>% mutate(rep=factor(rep, c("1", "2", "3", "4")))%>% mutate(time_replicate=factor(time_replicate,c("UI_1", "UI_2", "UI_3", "UI_4", "0h_1","0h_2","24h_1","24h_2","48h_1","48h_2","48h_3","48h_4")))

#def_list <- readr::read_tsv(filepath_def) %>% mutate(seq=factor(seq, c("H3p3", "H3K27ac","H3K4me3","H3K27me3")))  %>% 
#mutate(time1 = time, rep1=rep) %>% unite(time1,rep1,col="time_replicate")  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% mutate(type=factor(type,c("DoxPlus","DoxMinus"))) %>% mutate(rep=factor(rep, c("1", "2")))%>% mutate(time_replicate=factor(time_replicate,c("UI_1", "UI_2", "0h_1","0h_2","24h_1","24h_2","48h_1","48h_2")))


# add 20200617
groups <- c(
  "ATAC_UI_DoxMinus","ATAC_UI_DoxPlus",
  "ATAC_48h_DoxMinus","ATAC_48h_DoxPlus",
  
  "H3p3_UI_DoxMinus","H3p3_UI_DoxPlus",
  "H3p3_0h_DoxMinus","H3p3_0h_DoxPlus",
  "H3p3_24h_DoxMinus","H3p3_24h_DoxPlus",
  "H3p3_48h_DoxMinus","H3p3_48h_DoxPlus",
  
  "H3K27ac_UI_DoxMinus","H3K27ac_UI_DoxPlus",
  "H3K27ac_0h_DoxMinus","H3K27ac_0h_DoxPlus",
  "H3K27ac_24h_DoxMinus","H3K27ac_24h_DoxPlus",
  "H3K27ac_48h_DoxMinus","H3K27ac_48h_DoxPlus",
  
  "H3K4me3_UI_DoxMinus","H3K4me3_UI_DoxPlus",
  "H3K4me3_0h_DoxMinus","H3K4me3_0h_DoxPlus",
  "H3K4me3_24h_DoxMinus","H3K4me3_24h_DoxPlus",
  "H3K4me3_48h_DoxMinus","H3K4me3_48h_DoxPlus",
  
  "H3K27me3_UI_DoxMinus","H3K27me3_UI_DoxPlus",
  "H3K27me3_0h_DoxMinus","H3K27me3_0h_DoxPlus",
  "H3K27me3_24h_DoxMinus","H3K27me3_24h_DoxPlus",
  "H3K27me3_48h_DoxMinus","H3K27me3_48h_DoxPlus")

group_H3p3 <- c(
  "H3p3_UI_DoxMinus","H3p3_UI_DoxPlus",
  "H3p3_0h_DoxMinus","H3p3_0h_DoxPlus",
  "H3p3_24h_DoxMinus","H3p3_24h_DoxPlus",
  "H3p3_48h_DoxMinus","H3p3_48h_DoxPlus")

group_ATAC <- c(
  "ATAC_UI_DoxMinus","ATAC_UI_DoxPlus",
  "ATAC_48h_DoxMinus","ATAC_48h_DoxPlus")


samples <- c(
  "ATAC_UI_DoxMinus_1","ATAC_UI_DoxMinus_2","ATAC_UI_DoxMinus_3","ATAC_UI_DoxMinus_4",
  "ATAC_UI_DoxPlus_1","ATAC_UI_DoxPlus_2","ATAC_UI_DoxPlus_3","ATAC_UI_DoxPlus_4",
  "ATAC_48h_DoxMinus_1","ATAC_48h_DoxMinus_2","ATAC_48h_DoxMinus_3","ATAC_48h_DoxMinus_4",
  "ATAC_48h_DoxPlus_1","ATAC_48h_DoxPlus_2","ATAC_48h_DoxPlus_3","ATAC_48h_DoxPlus_4",
  
  "H3p3_UI_DoxMinus_1","H3p3_UI_DoxMinus_2","H3p3_UI_DoxPlus_1","H3p3_UI_DoxPlus_2",
  "H3p3_0h_DoxMinus_1","H3p3_0h_DoxMinus_2","H3p3_0h_DoxPlus_1","H3p3_0h_DoxPlus_2",
  "H3p3_24h_DoxMinus_1","H3p3_24h_DoxMinus_2","H3p3_24h_DoxPlus_1","H3p3_24h_DoxPlus_2",
  "H3p3_48h_DoxMinus_1","H3p3_48h_DoxMinus_2","H3p3_48h_DoxPlus_1","H3p3_48h_DoxPlus_2",
  
  "H3K27ac_UI_DoxMinus_1","H3K27ac_UI_DoxMinus_2","H3K27ac_UI_DoxPlus_1","H3K27ac_UI_DoxPlus_2",
  "H3K27ac_0h_DoxMinus_1","H3K27ac_0h_DoxMinus_2","H3K27ac_0h_DoxPlus_1","H3K27ac_0h_DoxPlus_2",
  "H3K27ac_24h_DoxMinus_1","H3K27ac_24h_DoxMinus_2","H3K27ac_24h_DoxPlus_1","H3K27ac_24h_DoxPlus_2",
  "H3K27ac_48h_DoxMinus_1","H3K27ac_48h_DoxMinus_2","H3K27ac_48h_DoxPlus_1","H3K27ac_48h_DoxPlus_2",
  
  "H3K4me3_UI_DoxMinus_1","H3K4me3_UI_DoxMinus_2","H3K4me3_UI_DoxPlus_1","H3K4me3_UI_DoxPlus_2",
  "H3K4me3_0h_DoxMinus_1","H3K4me3_0h_DoxMinus_2","H3K4me3_0h_DoxPlus_1","H3K4me3_0h_DoxPlus_2",
  "H3K4me3_24h_DoxMinus_1","H3K4me3_24h_DoxMinus_2","H3K4me3_24h_DoxPlus_1","H3K4me3_24h_DoxPlus_2",
  "H3K4me3_48h_DoxMinus_1","H3K4me3_48h_DoxMinus_2","H3K4me3_48h_DoxPlus_1","H3K4me3_48h_DoxPlus_2",
  
  "H3K27me3_UI_DoxMinus_1","H3K27me3_UI_DoxMinus_2","H3K27me3_UI_DoxPlus_1","H3K27me3_UI_DoxPlus_2",
  "H3K27me3_0h_DoxMinus_1","H3K27me3_0h_DoxMinus_2","H3K27me3_0h_DoxPlus_1","H3K27me3_0h_DoxPlus_2",
  "H3K27me3_24h_DoxMinus_1","H3K27me3_24h_DoxMinus_2","H3K27me3_24h_DoxPlus_1","H3K27me3_24h_DoxPlus_2",
  "H3K27me3_48h_DoxMinus_1","H3K27me3_48h_DoxMinus_2","H3K27me3_48h_DoxPlus_1","H3K27me3_48h_DoxPlus_2")

samples_ATAC <- c(
  "ATAC_UI_DoxMinus_1","ATAC_UI_DoxMinus_2","ATAC_UI_DoxMinus_3","ATAC_UI_DoxMinus_4",
  "ATAC_UI_DoxPlus_1","ATAC_UI_DoxPlus_2","ATAC_UI_DoxPlus_3","ATAC_UI_DoxPlus_4",
  "ATAC_48h_DoxMinus_1","ATAC_48h_DoxMinus_2","ATAC_48h_DoxMinus_3","ATAC_48h_DoxMinus_4",
  "ATAC_48h_DoxPlus_1","ATAC_48h_DoxPlus_2","ATAC_48h_DoxPlus_3","ATAC_48h_DoxPlus_4")

samples_H3p3 <- c(
  "H3p3_UI_DoxMinus_1","H3p3_UI_DoxMinus_2","H3p3_UI_DoxPlus_1","H3p3_UI_DoxPlus_2",
  "H3p3_0h_DoxMinus_1","H3p3_0h_DoxMinus_2","H3p3_0h_DoxPlus_1","H3p3_0h_DoxPlus_2",
  "H3p3_24h_DoxMinus_1","H3p3_24h_DoxMinus_2","H3p3_24h_DoxPlus_1","H3p3_24h_DoxPlus_2",
  "H3p3_48h_DoxMinus_1","H3p3_48h_DoxMinus_2","H3p3_48h_DoxPlus_1","H3p3_48h_DoxPlus_2")

samples_H3K27ac <- c(
  "H3K27ac_UI_DoxMinus_1","H3K27ac_UI_DoxMinus_2","H3K27ac_UI_DoxPlus_1","H3K27ac_UI_DoxPlus_2",
  "H3K27ac_0h_DoxMinus_1","H3K27ac_0h_DoxMinus_2","H3K27ac_0h_DoxPlus_1","H3K27ac_0h_DoxPlus_2",
  "H3K27ac_24h_DoxMinus_1","H3K27ac_24h_DoxMinus_2","H3K27ac_24h_DoxPlus_1","H3K27ac_24h_DoxPlus_2",
  "H3K27ac_48h_DoxMinus_1","H3K27ac_48h_DoxMinus_2","H3K27ac_48h_DoxPlus_1","H3K27ac_48h_DoxPlus_2")

samples_H3K4me3 <- c(
  "H3K4me3_UI_DoxMinus_1","H3K4me3_UI_DoxMinus_2","H3K4me3_UI_DoxPlus_1","H3K4me3_UI_DoxPlus_2",
  "H3K4me3_0h_DoxMinus_1","H3K4me3_0h_DoxMinus_2","H3K4me3_0h_DoxPlus_1","H3K4me3_0h_DoxPlus_2",
  "H3K4me3_24h_DoxMinus_1","H3K4me3_24h_DoxMinus_2","H3K4me3_24h_DoxPlus_1","H3K4me3_24h_DoxPlus_2",
  "H3K4me3_48h_DoxMinus_1","H3K4me3_48h_DoxMinus_2","H3K4me3_48h_DoxPlus_1","H3K4me3_48h_DoxPlus_2")

samples_H3K27me3 <- c(
  "H3K27me3_UI_DoxMinus_1","H3K27me3_UI_DoxMinus_2","H3K27me3_UI_DoxPlus_1","H3K27me3_UI_DoxPlus_2",
  "H3K27me3_0h_DoxMinus_1","H3K27me3_0h_DoxMinus_2","H3K27me3_0h_DoxPlus_1","H3K27me3_0h_DoxPlus_2",
  "H3K27me3_24h_DoxMinus_1","H3K27me3_24h_DoxMinus_2","H3K27me3_24h_DoxPlus_1","H3K27me3_24h_DoxPlus_2",
  "H3K27me3_48h_DoxMinus_1","H3K27me3_48h_DoxMinus_2","H3K27me3_48h_DoxPlus_1","H3K27me3_48h_DoxPlus_2")






f_sample <- function(x) x %>% mutate(sample=factor(sample, samples))
f_group <- function(x) x %>% mutate(group=factor(group, groups))

# filter(sample!="Doxminus_D48_ATAC_1") => filter((sample!="Doxminus_D48_ATAC_1")&(rep!="lot4")) (2020 0114修正)

#def_list <- def_list %>% f_sample %>% f_group

####
def_list_select <- def_list
def_list_select_0 <- def_list  %>% dplyr::select(-"file",-"multicov_No") %>% filter(seq=="ATAC")
def_list_select_1 <- def_list  %>% dplyr::select(-"file",-"multicov_No") %>% filter(seq=="H3p3")
def_list_select_2 <- def_list  %>% dplyr::select(-"file",-"multicov_No") %>% filter(seq=="H3K27ac")
def_list_select_3 <- def_list  %>% dplyr::select(-"file",-"multicov_No") %>% filter(seq=="H3K4me3")
def_list_select_4 <- def_list  %>% dplyr::select(-"file",-"multicov_No") %>% filter(seq=="H3K27me3")

#def_list_select <- def_list %>% filter(!!use) #使わないサンプルを削除(20190917)

#%>% 
#mutate(time1 = time, rep1=rep) %>% unite(time1,rep1,col="time_replicate") %>% #mutate(time_replicate=factor(time_replicate,c("UI_1","UI_2","D1","D48_lot2")))

# narrowpeak の場合 (型を指定) 200612modif 200615modif
#narrow_colnames <- c("chr","start","end","name","score","strand","singnalValue","pValue","qValue","peak","chr0","start0","end0")

merge_colnames <- c("chr","TSSstart","TSSend","ens_gene","score","strand","TSS","Start","End")

matome0 <- readr::read_tsv(filepath_summary, col_names = c(merge_colnames, def_list$sample))  %>% mutate_if(names(.) %in% c("start","end","TSS","Start","End",def_list$sample), as.integer)

matome0 %>% dplyr::select("chr","TSSstart","TSSend","ens_gene") %>% unique() # # A tibble: 105,166 x 4

#matome0 %>% group_by(chr,ens_gene,strand,Start,End) %>% unique() # # A tibble: 105,166 x 4

#matome <- matome0 %>% group_by(chr,start,end,Name) %>% dplyr::top_n(1, singnalValue)  %>% dplyr::top_n(1, peak) %>% dplyr::ungroup()

#matome  <- matome0 %>% filter_at(def_list$sample,any_vars(. > 0))

matome1 <- matome0 %>% gather("sample", "count", -("chr":"End")) %>% filter(sample %in% def_list_select$sample) %>% left_join(def_list_select, .,by = "sample")

# %>% dplyr::select(-"score",-"strand",-"singnalValue",-"pValue",-"qValue",-"peak")

#---- 確認 ----#
matome0 %>% nrow()

filename <- gsub(".txt","__count.csv",basename(filepath_summary))  #geneにつき複数領域
matome0 %>% readr::write_csv(filename)
print(filename)

#filename <- gsub(".txt","__select_nameonly.csv",basename(filepath_def))
#matome %>% dplyr::select("chr","start","end","name") %>% readr::write_csv(filename)
#print(filename)

#matome %>% dplyr::select("chr","start","end","name","score","Name","strand","singnalValue","pValue","qValue","peak","chr0","start0","end0") %>% readr::write_csv(filename)
#print(filename)

#matome00 <- matome0 %>% group_by(chr,start,end,Name) %>% summarise(max_score=max(score), paste(Name, collapse="")) #dplyr::top_n(score,1)
  
#matome <- matome0 %>% filter(name %in% matome00$name)

#matome %>% mutate(name1=name) %>% filter(grepl("[a-u]$",name)) %>% mutate(Name=gsub("[a-u]$","",name))


#mat_select <- matome %>% dplyr::select(chr,start,end,name,Name) 
#annotate_bed <- partial(right_join,mat_select,by="ens_gene") #2gunで使う

```


```{r select first exon TSS 200706}
#matome1_number <- matome1 %>% mutate(position = row_number())
#matome1_plus <- matome1_number %>% filter(strand=="+")
#matome1_minus <- matome1_number %>% filter(strand=="-")

matome0_number <- matome0 %>% mutate(position = row_number())
nrow(matome0_number)

matome0_plus <- matome0_number %>% filter(strand=="+")
matome0_minus <- matome0_number %>% filter(strand=="-")

matome0_plus_o <- matome0_plus %>% group_by(chr,ens_gene) %>% dplyr::top_n(-1,TSS) #低
matome0_minus_o <- matome0_minus %>% group_by(chr,ens_gene) %>% dplyr::top_n(1,TSS) #高

matome0_o <- dplyr::bind_rows(matome0_plus_o, matome0_minus_o) %>% arrange(position)
nrow(matome0_o)



##----- 確認 ---------##
colnames(matome0_o) 
nrow(matome0_o)

filename <- gsub(".txt","__count_firstTSS.csv",basename(filepath_summary)) #geneにつき1領域
matome0_o %>% readr::write_csv(filename)
print(filename)

##--------------------##

#matome0_plus %>% filter(ens_gene =="ENSMUSG00000000037") %>% dplyr::select(TSSstart,position)
#matome0_plus_o %>% filter(ens_gene =="ENSMUSG00000000037") %>% dplyr::select(TSSstart,position)

#%>% group_by(chr,TSSstart,TSSend,ens_gene,score,strand,TSS,Start,End,seq) 

#matome1_number %>% filter(!(strand=="+"|strand=="-"))

```




```{r select 200715 notMT}
matome0_s <- matome0_o %>%  dplyr::select(chr,ens_gene,TSSstart,TSSend,score,strand,TSS,Start,End,position,all_of(samples)) %>% filter(chr!="chrM")
nrow(matome0_s)

#matome0_s <- matome0_o %>% filter(ens_gene %in% FC_rank_all_BRBlist$ens_gene) %>% dplyr::select(chr,ens_gene,TSSstart,TSSend,score,strand,TSS,Start,End,position,all_of(samples))
nrow(matome0_s)

#matome0_s1 <- matome0_s %>% left_join(FC_rank_all_BRBlist %>% dplyr::select(ens_gene,log2FoldChange,Rank))

matome5 <- matome0_s %>% dplyr::select(chr,ens_gene,position,all_of(samples)) %>% ungroup()


##----- 確認 ---------##
colnames(matome0_s) 
nrow(matome0_s)

filename <- gsub(".txt","__count_firstTSS_select.csv",basename(filepath_summary))  #geneにつき1領域、かつBRBでnormalized countがあるもの 
matome0_s %>% readr::write_csv(filename)
print(filename)

##--------------------##

annotate_TSS <- partial(right_join,dplyr::select(matome0_s,chr,ens_gene,TSSstart,TSSend,score,strand,TSS,Start,End,position),by="ens_gene") #2gunで使う
```


## normalized count

separate matrix


```{r setup matrix Deseq2 part1 200707}

nrow(matome5)
colnames(matome5)

X <- matome5 %>% dplyr::select(all_of(samples)) %>% as.matrix
rownames(X) <- matome5$ens_gene

###--- DESeq2によりnormalized count (必要なサンプルのみ) ---###
model<- ~group
dds_ATAC <- DESeq2::DESeqDataSetFromMatrix(X[,def_list_select_0$sample],def_list_select_0,model) #ATAC
dds_H3p3 <- DESeq2::DESeqDataSetFromMatrix(X[,def_list_select_1$sample],def_list_select_1,model) #H3p3
dds_H3K27ac <- DESeq2::DESeqDataSetFromMatrix(X[,def_list_select_2$sample],def_list_select_2,model) #H3K27ac
dds_H3K4me3 <- DESeq2::DESeqDataSetFromMatrix(X[,def_list_select_3$sample],def_list_select_3,model) #H3K4me3
dds_H3K27me3 <- DESeq2::DESeqDataSetFromMatrix(X[,def_list_select_4$sample],def_list_select_4,model) #H3K27me3

```

#### Fit model

```{r deseq2 2gun}
#model_2gun <- ~group
#dds_2gun <- DESeq2::DESeqDataSetFromMatrix(X[,def_list_select$sample],def_list_select,model_2gun)
dds_ATAC <- DESeq2::DESeq(dds_ATAC)
dds_H3p3 <- DESeq2::DESeq(dds_H3p3)
dds_H3K27ac <- DESeq2::DESeq(dds_H3K27ac)
dds_H3K4me3 <- DESeq2::DESeq(dds_H3K4me3)
dds_H3K27me3 <- DESeq2::DESeq(dds_H3K27me3)

#keep <- rowSums(counts(dds)) >= 10 #low countは削る方法
#dds <- dds[keep,] #low countは削る方法

```


#### Diagnostics plot

```{r diagnostics 2gun,fig.width=7,fig.height=5}
DESeq2::sizeFactors(dds_ATAC) %>%
  {tibble(sample=names(.),sizeFactor=.)} %>%
  ggplot(aes(sample,sizeFactor)) + theme_minimal() +
  geom_bar(stat="identity") + coord_flip()
DESeq2::plotDispEsts(dds_ATAC)

DESeq2::sizeFactors(dds_H3p3) %>%
  {tibble(sample=names(.),sizeFactor=.)} %>%
  ggplot(aes(sample,sizeFactor)) + theme_minimal() +
  geom_bar(stat="identity") + coord_flip()
DESeq2::plotDispEsts(dds_H3p3)

DESeq2::sizeFactors(dds_H3K27ac) %>%
  {tibble(sample=names(.),sizeFactor=.)} %>%
  ggplot(aes(sample,sizeFactor)) + theme_minimal() +
  geom_bar(stat="identity") + coord_flip()
DESeq2::plotDispEsts(dds_H3K27ac)

DESeq2::sizeFactors(dds_H3K4me3) %>%
  {tibble(sample=names(.),sizeFactor=.)} %>%
  ggplot(aes(sample,sizeFactor)) + theme_minimal() +
  geom_bar(stat="identity") + coord_flip()
DESeq2::plotDispEsts(dds_H3K4me3)

DESeq2::sizeFactors(dds_H3K27me3) %>%
  {tibble(sample=names(.),sizeFactor=.)} %>%
  ggplot(aes(sample,sizeFactor)) + theme_minimal() +
  geom_bar(stat="identity") + coord_flip()
DESeq2::plotDispEsts(dds_H3K27me3)
```


normalized count listを書き出し

```{r setup matrix Deseq2 part2 200707}

# ATAC
dds_ATAC <- DESeq2::estimateSizeFactors(dds_ATAC)
norm_ATAC <- DESeq2::counts(dds_ATAC,normalized=TRUE) #DEGを取った後のクラスタリングに使う。

normalizedcount_ATAC <- as.data.frame(norm_ATAC) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_0$sample))
filename <- paste(csvfilepath,"_normcount_ATAC.csv",sep="_")
print(filename)
readr::write_csv(normalizedcount_ATAC, filename)
nrow(normalizedcount_ATAC)
ncol(normalizedcount_ATAC)

norm_gene_ATAC <- normalizedcount_ATAC %>% inner_join(ensemble) %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(def_list_select_0$sample))
filename <- paste(csvfilepath,"_normcount_ATAC_genename.csv",sep="_")
print(filename)
readr::write_csv(norm_gene_ATAC, filename)
nrow(norm_gene_ATAC)
ncol(norm_gene_ATAC)


# H3p3
dds_H3p3 <- DESeq2::estimateSizeFactors(dds_H3p3)
norm_H3p3 <- DESeq2::counts(dds_H3p3,normalized=TRUE) #DEGを取った後のクラスタリングに使う。

normalizedcount_H3p3 <- as.data.frame(norm_H3p3) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_1$sample))
filename <- paste(csvfilepath,"_normcount_H3p3.csv",sep="_")
print(filename)
readr::write_csv(normalizedcount_H3p3, filename)
nrow(normalizedcount_H3p3)
ncol(normalizedcount_H3p3)

norm_gene_H3p3 <- normalizedcount_H3p3 %>% inner_join(ensemble) %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(def_list_select_1$sample))
filename <- paste(csvfilepath,"_normcount_H3p3_genename.csv",sep="_")
print(filename)
readr::write_csv(norm_gene_H3p3, filename)
nrow(norm_gene_H3p3)
ncol(norm_gene_H3p3)


# H3K27ac
dds_H3K27ac <- DESeq2::estimateSizeFactors(dds_H3K27ac)
norm_H3K27ac <- DESeq2::counts(dds_H3K27ac,normalized=TRUE) #DEGを取った後のクラスタリングに使う。

normalizedcount_H3K27ac <- as.data.frame(norm_H3K27ac) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_2$sample))
filename <- paste(csvfilepath,"_normcount_H3K27ac.csv",sep="_")
print(filename)
readr::write_csv(normalizedcount_H3K27ac, filename)
nrow(normalizedcount_H3K27ac)
ncol(normalizedcount_H3K27ac)

norm_gene_H3K27ac <- normalizedcount_H3K27ac %>% inner_join(ensemble) %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(def_list_select_2$sample))
filename <- paste(csvfilepath,"_normcount_H3K27ac_genename.csv",sep="_")
print(filename)
readr::write_csv(norm_gene_H3K27ac, filename)
nrow(norm_gene_H3K27ac)
ncol(norm_gene_H3K27ac)

# H3K4me3
dds_H3K4me3 <- DESeq2::estimateSizeFactors(dds_H3K4me3)
norm_H3K4me3 <- DESeq2::counts(dds_H3K4me3,normalized=TRUE) #DEGを取った後のクラスタリングに使う。

normalizedcount_H3K4me3 <- as.data.frame(norm_H3K4me3) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_3$sample))
filename <- paste(csvfilepath,"_normcount_H3K4me3.csv",sep="_")
print(filename)
readr::write_csv(normalizedcount_H3K4me3, filename)
nrow(normalizedcount_H3K4me3)
ncol(normalizedcount_H3K4me3)

norm_gene_H3K4me3 <- normalizedcount_H3K4me3 %>% inner_join(ensemble) %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(def_list_select_3$sample))
filename <- paste(csvfilepath,"_normcount_H3K4me3_genename.csv",sep="_")
print(filename)
readr::write_csv(norm_gene_H3K4me3, filename)
nrow(norm_gene_H3K4me3)
ncol(norm_gene_H3K4me3)

# H3K27me3
dds_H3K27me3 <- DESeq2::estimateSizeFactors(dds_H3K27me3)
norm_H3K27me3 <- DESeq2::counts(dds_H3K27me3,normalized=TRUE) #DEGを取った後のクラスタリングに使う。

normalizedcount_H3K27me3 <- as.data.frame(norm_H3K27me3) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_4$sample))
filename <- paste(csvfilepath,"_normcount_H3K27me3.csv",sep="_")
print(filename)
readr::write_csv(normalizedcount_H3K27me3, filename)
nrow(normalizedcount_H3K27me3)
ncol(normalizedcount_H3K27me3)

norm_gene_H3K27me3 <- normalizedcount_H3K27me3 %>% inner_join(ensemble) %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(def_list_select_4$sample))
filename <- paste(csvfilepath,"_normcount_H3K27me3_genename.csv",sep="_")
print(filename)
readr::write_csv(norm_gene_H3K27me3, filename)
nrow(norm_gene_H3K27me3)
ncol(norm_gene_H3K27me3)


# bind norm count
normalizedcount <- normalizedcount_ATAC %>% inner_join(normalizedcount_H3p3) %>% inner_join(normalizedcount_H3K27ac) %>% inner_join(normalizedcount_H3K4me3) %>% inner_join(normalizedcount_H3K27me3)
norm_gene <- norm_gene_ATAC %>% inner_join(norm_gene_H3p3) %>% inner_join(norm_gene_H3K27ac) %>% inner_join(norm_gene_H3K4me3) %>% inner_join(norm_gene_H3K27me3)

print(norm_gene)
colnames(norm_gene)

filename <- paste(csvfilepath,"_normcount.csv",sep="_")
print(filename)
readr::write_csv(normalizedcount, filename)

filename <- paste(csvfilepath,"_normcount_genename.csv",sep="_")
print(filename)
readr::write_csv(norm_gene, filename)


```

size factors を書き出し

```{r setup matrix Deseq2 part3 200707}

filename <- paste(csvfilepath,"_sizefactors_ATAC.csv",sep="_")
print(filename)
as.data.frame(DESeq2::sizeFactors(dds_ATAC))  %>% tibble::rownames_to_column("sample") %>% readr::write_csv(filename)

filename <- paste(csvfilepath,"_sizefactors_H3p3.csv",sep="_")
print(filename)
as.data.frame(DESeq2::sizeFactors(dds_H3p3))  %>% tibble::rownames_to_column("sample") %>% readr::write_csv(filename)

filename <- paste(csvfilepath,"_sizefactors_H3K27ac.csv",sep="_")
print(filename)
as.data.frame(DESeq2::sizeFactors(dds_H3K27ac))  %>% tibble::rownames_to_column("sample") %>% readr::write_csv(filename)

filename <- paste(csvfilepath,"_sizefactors_H3K4me3.csv",sep="_")
print(filename)
as.data.frame(DESeq2::sizeFactors(dds_H3K4me3))  %>% tibble::rownames_to_column("sample") %>% readr::write_csv(filename)

filename <- paste(csvfilepath,"_sizefactors_H3K27me3.csv",sep="_")
print(filename)
as.data.frame(DESeq2::sizeFactors(dds_H3K27me3))  %>% tibble::rownames_to_column("sample") %>% readr::write_csv(filename)

```

vst => z score

```{r setup matrix Deseq2 part4 200707}

vsd_ATAC <- DESeq2::vst(dds_ATAC) #normalized countが入っている。(vstかrlog)
Xd_ATAC <- SummarizedExperiment::assay(vsd_ATAC) # 全て選択(200326) 20190920を元に (191024)
Xs_ATAC <- Xd_ATAC %>% t %>% scale %>% t

vsd_H3p3 <- DESeq2::vst(dds_H3p3) #normalized countが入っている。(vstかrlog)
Xd_H3p3 <- SummarizedExperiment::assay(vsd_H3p3) # 全て選択(200326) 20190920を元に (191024)
Xs_H3p3 <- Xd_H3p3 %>% t %>% scale %>% t

vsd_H3K27ac <- DESeq2::vst(dds_H3K27ac) #normalized countが入っている。(vstかrlog)
Xd_H3K27ac <- SummarizedExperiment::assay(vsd_H3K27ac) # 全て選択(200326) 20190920を元に (191024)
Xs_H3K27ac <- Xd_H3K27ac %>% t %>% scale %>% t

vsd_H3K4me3 <- DESeq2::vst(dds_H3K4me3) #normalized countが入っている。(vstかrlog)
Xd_H3K4me3 <- SummarizedExperiment::assay(vsd_H3K4me3) # 全て選択(200326) 20190920を元に (191024)
Xs_H3K4me3 <- Xd_H3K4me3 %>% t %>% scale %>% t

vsd_H3K27me3 <- DESeq2::vst(dds_H3K27me3) #normalized countが入っている。(vstかrlog)
Xd_H3K27me3 <- SummarizedExperiment::assay(vsd_H3K27me3) # 全て選択(200326) 20190920を元に (191024)
Xs_H3K27me3 <- Xd_H3K27me3 %>% t %>% scale %>% t

vsdtrans_ATAC <- as.data.frame(Xd_ATAC) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_0$sample))
filename <- paste(csvfilepath,"_vstrans_ATAC.csv",sep="_")
readr::write_csv(vsdtrans_ATAC, filename)
nrow(vsdtrans_ATAC)
ncol(vsdtrans_ATAC)

vsdtrans_H3p3 <- as.data.frame(Xd_H3p3) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_1$sample))
filename <- paste(csvfilepath,"_vstrans_H3p3.csv",sep="_")
readr::write_csv(vsdtrans_H3p3, filename)
nrow(vsdtrans_H3p3)
ncol(vsdtrans_H3p3)

vsdtrans_H3K27ac <- as.data.frame(Xd_H3K27ac) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_2$sample))
filename <- paste(csvfilepath,"_vstrans_H3K27ac.csv",sep="_")
readr::write_csv(vsdtrans_H3K27ac, filename)
nrow(vsdtrans_H3K27ac)
ncol(vsdtrans_H3K27ac)

vsdtrans_H3K4me3 <- as.data.frame(Xd_H3K4me3) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_3$sample))
filename <- paste(csvfilepath,"_vstrans_H3K4me3.csv",sep="_")
readr::write_csv(vsdtrans_H3K4me3, filename)
nrow(vsdtrans_H3K4me3)
ncol(vsdtrans_H3K4me3)

vsdtrans_H3K27me3 <- as.data.frame(Xd_H3K27me3) %>% tibble::rownames_to_column("ens_gene") %>% as_tibble %>% dplyr::select("ens_gene", all_of(def_list_select_4$sample))
filename <- paste(csvfilepath,"_vstrans_H3K27me3.csv",sep="_")
readr::write_csv(vsdtrans_H3K27me3, filename)
nrow(vsdtrans_H3K27me3)
ncol(vsdtrans_H3K27me3)

zscore_ATAC <- Xs_ATAC  %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble
zscore_H3p3 <- Xs_H3p3  %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble
zscore_H3K27ac <- Xs_H3K27ac  %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble
zscore_H3K4me3 <- Xs_H3K4me3  %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble
zscore_H3K27me3 <- Xs_H3K27me3  %>% as.data.frame() %>% tibble::rownames_to_column("ens_gene") %>% as_tibble

zscore <- zscore_ATAC %>% inner_join(zscore_H3p3) %>% inner_join(zscore_H3K27ac) %>% inner_join(zscore_H3K4me3) %>% inner_join(zscore_H3K27me3)
readr::write_csv(zscore, paste(csvfilepath,"_zscore_All.csv",sep="_"))

zscore_type <- zscore  %>% inner_join(ensemble) %>% dplyr::select("ens_gene","ext_gene", "biotype","chr", all_of(def_list_select$sample))


```




### norm後の分布
normalized count, zscore の分布
(20191017修正)

```{r allnormalizedcount, fig.width=7,fig.height=6}

norm_plotlist_all <- normalizedcount %>% gather("sample", "normalized",-(ens_gene)) %>% inner_join(def_list_select, by = "sample")
norm_plot_all <- norm_plotlist_all %>% 
ggplot(aes(time_replicate,normalized,group=time_replicate,colour=type))+geom_violin(scale="count")+geom_boxplot(width=.1,)+facet_wrap(~seq*type,ncol=2)+theme_bw()+ theme(strip.text = element_text(size = 12),axis.text.x = element_text(hjust = 0.5,vjust = 0.5)) + ggtitle(paste(folder_name, "(all, normalized count)")) + scale_y_log10(limits = c(0.1,NA)) 

#+ggsci::scale_color_d3("category20")

print(norm_plot_all)



ggsave(plot=norm_plot_all,file="./NormCount.pdf", width = 20, height = 15, dpi = 360, limitsize = FALSE)

```



## QC

### Total reads

20191016修正, 20200623修正

```{r totalReads, fig.width=7,fig.height=8}

chr_chart <- matome5 %>% dplyr::select(ens_gene,chr) %>% group_by(ens_gene,chr) %>% summarize() %>% ungroup() 
#chr_chart <- bedfile %>% dplyr::select(ens_gene,chr)  %>% group_by(ens_gene,chr) %>% summarize() %>% ungroup() #20191016修正

#> right_join(chr_chart, matome5, by="ens_gene")
# A tibble: 55,456 x 14

#bychr<- left_join(matome5, chr_chart, by="ens_gene") %>% dplyr::select(-(ens_gene)) %>%
#  gather("sample","count",-chr) %>%
#  group_by(chr,sample) %>% summarise(total=sum(count)) %>% ungroup

#mat <- left_join(matome5, chr_chart, by="ens_gene")

#bychr <- mat %>% dplyr::select(-(ens_gene)) %>%
#  gather("sample","count",-chr) %>%
#  group_by(chr,sample) %>% summarise(total=sum(count))  %>% f_sample() %>% ungroup

bychr <- matome5 %>% dplyr::select(chr,def_list_select$sample) %>%
  gather("sample","count",-chr) %>%
  group_by(chr,sample) %>% summarise(total=sum(count))  %>% f_sample() %>% ungroup

ggplot(bychr,aes(reorder(sample,dplyr::desc(sample)),total/1e6,fill=chr)) +
  theme_linedraw() + geom_bar(stat="identity") + coord_flip() +
  xlab("sample") + ylab("million reads") + ggsci::scale_fill_igv() +
  scale_x_discrete(limits = rev(levels(sample)))




```

### normalized count

```{r normalizedcount, fig.width=7,fig.height=8}

mat_normcount <- left_join(normalizedcount, chr_chart, by="ens_gene")

bychr_mat_normcount <- mat_normcount %>% dplyr::select(-(ens_gene)) %>%
  gather("sample","normcount",-chr) %>%
  group_by(chr,sample) %>% summarise(normtotal=sum(normcount)) %>% f_sample() %>% ungroup

ggplot(bychr_mat_normcount,aes(reorder(sample,dplyr::desc(sample)),normtotal/1e6,fill=chr)) +
  theme_linedraw() + geom_bar(stat="identity") + coord_flip() +
  xlab("sample") + ylab("normalized counts (million reads)") + ggsci::scale_fill_igv() +
  scale_x_discrete(limits = rev(levels(sample)))

```

### Correlations

drop rows with all 0 -> +1/2 -> geom.scale -> log -> Pearson's

```{r makemat, fig.width=11, fig.height=11}
#matf_Correlation <- mat %>% filter(chr!="chrM") %>% dplyr::select(-"chr") %>% filter_at(-(1),any_vars(. > 0))

matf_Correlation <- matome5 %>% dplyr::select(chr,def_list_select$sample) %>% filter(chr!="chrM") %>% dplyr::select(-"chr") %>% filter_at(-(1),any_vars(. > 0))

X_Correlation <- matf_Correlation %>% dplyr::select(-(1)) %>% as.matrix
rownames(X_Correlation) <- matf_Correlation$ens_gene
lX_Correlation <- log(gscale(X_Correlation+0.5))
R <- cor(lX_Correlation); diag(R) <- NA
pheatmap::pheatmap(R,color=viridis::viridis(256))

#X <- matf %>% dplyr::select(-(1:4)) %>% as.matrix
#rownames(X) <- matf$ens_gene
#lX <- log(gscale(X+0.5))
#R <- cor(lX); diag(R) <- NA
#pheatmap::pheatmap(R,color=viridis::viridis(256))

###--- DESeq2によりnormalized count ---###
#model<- ~group
#dds_Correlation <- DESeq2::DESeqDataSetFromMatrix(X_Correlation[,def_list_select$sample],def_list_select,model) #必要なサンプルのみ
#dds_Correlation <- DESeq2::DESeqDataSetFromMatrix(X_Correlation[,def_list$sample],def_list,model)
#dds_Correlation <- DESeq2::estimateSizeFactors(dds_Correlation)
#norm_Correlation <- DESeq2::counts(dds_Correlation,normalized=TRUE) #normにnarmalized countが入る。

```

### Dimension reduction

```{r PCA,fig.width=4,fig.height=3}
# set scale=TRUE if the patterns (not level) is the matter
p <- prcomp(t(lX_Correlation[rank(-apply(lX_Correlation,1,var)) <= ntop,]),scale=scalerows,center=TRUE)
screeplot(p,las=2,main="Importance")
print(summary(p)$imp[,seq(min(10,ncol(X_Correlation)))])
```

```{r makescoreDF}
label_Correlation <- def_list_select %>% filter(sample %in% colnames(X_Correlation))
df <- data.frame(p$x) %>% as_tibble(rownames="sample") %>%
  inner_join(label_Correlation,.)

print(df)
```


QC 終了

### Calculate log2 FC



```{r result 2gun set}

#------- setting -------#
fdr <- 0.1 # acceptable false discovery rate (固定)
lfcthreth <- log2(1) # threshold in abs(log2FC)
# controls should be placed in the right

plot_title1 <- "2gun"


contrast_H3p3 <- list(
  #Intercept = list("Intercept"),
  group_H3p3_UI_Doxplus_vs_minus = c("group", "H3p3_UI_DoxPlus","H3p3_UI_DoxMinus"),
  group_H3p3_0h_Doxplus_vs_minus = c("group","H3p3_0h_DoxPlus", "H3p3_0h_DoxMinus"),
  group_H3p3_24h_Doxplus_vs_minus = c("group","H3p3_24h_DoxPlus", "H3p3_24h_DoxMinus"),
  group_H3p3_48h_Doxplus_vs_minus = c("group","H3p3_48h_DoxPlus", "H3p3_48h_DoxMinus")
)



contrast_H3K4me3 <- list(
  #Intercept = list("Intercept"),
  group_H3K4me3_UI_Doxplus_vs_minus = c("group", "H3K4me3_UI_DoxPlus","H3K4me3_UI_DoxMinus"),
  group_H3K4me3_0h_Doxplus_vs_minus = c("group","H3K4me3_0h_DoxPlus", "H3K4me3_0h_DoxMinus"),
  group_H3K4me3_24h_Doxplus_vs_minus = c("group","H3K4me3_24h_DoxPlus", "H3K4me3_24h_DoxMinus"),
  group_H3K4me3_48h_Doxplus_vs_minus = c("group","H3K4me3_48h_DoxPlus", "H3K4me3_48h_DoxMinus")
)

contrast_H3K27ac <- list(
  #Intercept = list("Intercept"),
  group_H3K27ac_UI_Doxplus_vs_minus = c("group", "H3K27ac_UI_DoxPlus","H3K27ac_UI_DoxMinus"),
  group_H3K27ac_0h_Doxplus_vs_minus = c("group","H3K27ac_0h_DoxPlus", "H3K27ac_0h_DoxMinus"),
  group_H3K27ac_24h_Doxplus_vs_minus = c("group","H3K27ac_24h_DoxPlus", "H3K27ac_24h_DoxMinus"),
  group_H3K27ac_48h_Doxplus_vs_minus = c("group","H3K27ac_48h_DoxPlus", "H3K27ac_48h_DoxMinus")
)


contrast_H3K27me3 <- list(
  #Intercept = list("Intercept"),
  group_H3K27me3_UI_Doxplus_vs_minus = c("group", "H3K27me3_UI_DoxPlus","H3K27me3_UI_DoxMinus"),
  group_H3K27me3_0h_Doxplus_vs_minus = c("group","H3K27me3_0h_DoxPlus", "H3K27me3_0h_DoxMinus"),
  group_H3K27me3_24h_Doxplus_vs_minus = c("group","H3K27me3_24h_DoxPlus", "H3K27me3_24h_DoxMinus"),
  group_H3K27me3_48h_Doxplus_vs_minus = c("group","H3K27me3_48h_DoxPlus", "H3K27me3_48h_DoxMinus")
)

contrast_ATAC <- list(
  #Intercept = list("Intercept"),
  group_ATAC_UI_Doxplus_vs_minus = c("group", "ATAC_UI_DoxPlus","ATAC_UI_DoxMinus"),
  group_ATAC_48h_Doxplus_vs_minus = c("group","ATAC_48h_DoxPlus", "ATAC_48h_DoxMinus")
)

# BRB
#  group_UI_Doxplus_vs_minus = c("group", "BRB_UI_DoxPlus", "BRB_UI_DoxMinus"),
#  group_0h_Doxplus_vs_minus = c("group", "BRB_0h_DoxPlus", "BRB_0h_DoxMinus"),
#  group_24h_Doxplus_vs_minus = c("group", "BRB_24h_DoxPlus", "BRB_24h_DoxMinus"),
#  group_48h_Doxplus_vs_minus = c("group", "BRB_48h_DoxPlus", "BRB_48h_DoxMinus")
  


#-----------------------#

#-------------- 自動 --------------------------------------------------#
#-- ファイル名 の設定 ---#
#folder_name_plot0 <- paste(".",folder_name, paste(folder_name,plot_title1,sep="_"),"",sep="/")
#folder_name_plot_path <- paste(folder_name_plot0,paste(folder_name,plot_title1,"",sep="_"),sep="")

```


log2 FC のみ計算 (DEGは特に出さない)

```{r extractRes}


res_H3p3 <- mapply(function(x)
  DESeq2::results(dds_H3p3,x,lfcThreshold=lfcthreth,alpha=fdr)
,contrast_H3p3)

res_H3K4me3 <- mapply(function(x)
  DESeq2::results(dds_H3K4me3,x,lfcThreshold=lfcthreth,alpha=fdr)
,contrast_H3K4me3)

res_H3K27ac <- mapply(function(x)
  DESeq2::results(dds_H3K27ac,x,lfcThreshold=lfcthreth,alpha=fdr)
,contrast_H3K27ac)


res_H3K27me3 <- mapply(function(x)
  DESeq2::results(dds_H3K27me3,x,lfcThreshold=lfcthreth,alpha=fdr)
,contrast_H3K27me3)


res_ATAC <- mapply(function(x)
  DESeq2::results(dds_ATAC,x,lfcThreshold=lfcthreth,alpha=fdr)
,contrast_ATAC)



print(fdr)

```

```{r extractRes2}

re_H3p3_all <- map(res_H3p3,as_tibble,rownames="ens_gene") %>%
  tibble(aspect=factor(names(.),names(.)),data=.) %>%
  mutate(data=map(data,annotate)) %>%
  unnest(cols = "data")

re_H3K4me3_all <- map(res_H3K4me3,as_tibble,rownames="ens_gene") %>%
  tibble(aspect=factor(names(.),names(.)),data=.) %>%
  mutate(data=map(data,annotate)) %>%
  unnest(cols = "data")

re_H3K27ac_all <- map(res_H3K27ac,as_tibble,rownames="ens_gene") %>%
  tibble(aspect=factor(names(.),names(.)),data=.) %>%
  mutate(data=map(data,annotate)) %>%
  unnest(cols = "data")

re_H3K27me3_all <- map(res_H3K27me3,as_tibble,rownames="ens_gene") %>%
  tibble(aspect=factor(names(.),names(.)),data=.) %>%
  mutate(data=map(data,annotate)) %>%
  unnest(cols = "data")

re_ATAC_all <- map(res_ATAC,as_tibble,rownames="ens_gene") %>%
  tibble(aspect=factor(names(.),names(.)),data=.) %>%
  mutate(data=map(data,annotate)) %>%
  unnest(cols = "data")


filename <- paste("./2gun/",csvfilepath,"_resultsall_fdr0p1_H3p3.csv",sep="")
print(filename)
readr::write_csv(re_H3p3_all,filename)
nrow(re_H3p3_all)

filename <- paste("./2gun/",csvfilepath,"_resultsall_fdr0p1_H3K4me3.csv",sep="")
print(filename)
readr::write_csv(re_H3K4me3_all,filename)
nrow(re_H3K4me3_all)

filename <- paste("./2gun/",csvfilepath,"_resultsall_fdr0p1_H3K27ac.csv",sep="")
print(filename)
readr::write_csv(re_H3K27ac_all,filename)
nrow(re_H3K27ac_all)

filename <- paste("./2gun/",csvfilepath,"_resultsall_fdr0p1_H3K27me3.csv",sep="")
print(filename)
readr::write_csv(re_H3K27me3_all,filename)
nrow(re_H3K27me3_all)

filename <- paste("./2gun/",csvfilepath,"_resultsall_fdr0p1_ATAC.csv",sep="")
print(filename)
readr::write_csv(re_ATAC_all,filename)
nrow(re_ATAC_all)

```




----------------------

## Clustering (all genes)

clustering H3.3

```{r clustering H3p3 all genes kemans4, fig.width=6,fig.height=5}
#20191205修正と作成



cluster_number <- 6


##--------- clustering -----------#
set.seed(3)

# H3.3で
 
zscore_H3p3_s <- zscore_type %>% dplyr::select("ens_gene",all_of(def_list_select_1$sample))  %>% filter(across(where(is_double), ~ (.x != 0)|(.x == 0)))
#zscore_H3p3_s <- zscore_H3p3 %>% filter(across(where(is_double), ~ (.x != 0)|(.x == 0)))

nrow(zscore_type)
nrow(zscore_H3p3_s)

Xs_H3p3 <- zscore_H3p3_s %>% dplyr::select(-ens_gene) %>% as.matrix()
rownames(Xs_H3p3) <- zscore_H3p3_s$ens_gene

km_allH3p3 <- kmeans(Xs_H3p3,cluster_number,nstart = 25,algorithm = "Lloyd")
kmc_allH3p3 <- km_allH3p3$centers %>% as_tibble(rownames="cluster") %>% gather(sample,val,-cluster) %>% inner_join(def_list_select)

kmc_allH3p3_group <- kmc_allH3p3

#kmc_LRT_group <- kmc_LRT %>% mutate(growth=factor(growth, c("UI","Diff0h","Diff24h","Diff48h"))) %>% mutate(type=factor(type, c("Doxplus","Doxminus")))

#kmc_LRT_group <- kmc_LRT_group %>% mutate(time=case_when(growth=="UI" ~"UI",growth=="Diff0h"~"0h",growth=="Diff24h"~"24h",growth=="Diff48h"~"48h",TRUE~"error"))
#kmc_LRT_group <- kmc_LRT_group %>% mutate(time=factor(time, c("UI","0h","24h","48h")))

#gggglabel <- paste("k-means: Total",nrow(Xs_H3p3),"[1]",km_allH3p3$size[1],"[2]",km_allH3p3$size[2],"[3]",km_allH3p3$size[3],"[4]",km_allH3p3$size[4],"[5]",km_allH3p3$size[5],"[6]",km_allH3p3$size[6],sep=" ")

gggglabel <- paste("Original",nrow(zscore_type),"H3.3 k-means: Total",nrow(Xs_H3p3),"[1]",km_allH3p3$size[1],"[2]",km_allH3p3$size[2],"[3]",km_allH3p3$size[3],"[4]",km_allH3p3$size[4],"[5]",km_allH3p3$size[5],"[6]",km_allH3p3$size[6],sep=" ")

#------- size -------#

print(km_allH3p3$size) 
#rrres_allH3p3 <- km_allH3p3$cluster %>% tibble(ens_gene=names(.),cluster=.) %>% left_join(zscore_type_clus3,.) %>% arrange(cluster) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,cluster,all_of(def_list_select$sample))

rrres_allH3p3 <- km_allH3p3$cluster %>% tibble(ens_gene=names(.),cluster=.) %>% left_join(.,zscore_type) %>% arrange(cluster) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,cluster,all_of(def_list_select$sample))

#rrres_allH3p3 <- km_allH3p3$cluster %>% tibble(ens_gene=names(.),cluster=.) %>% left_join(zscore_type,.) %>% arrange(cluster) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,cluster,all_of(def_list_select$sample))

#rrres_LRT <- km_LRT$cluster %>% tibble(ens_gene=names(.),cluster=.) %>% right_join(e2g,.) %>% arrange(cluster)

file_path <- paste("./H3p3allcluster/", csvfilepath, "_kmeans_cluster.csv",sep="") 
readr::write_csv(rrres_allH3p3,file_path)

##------- PCA -------#

pcacluster_save <- prcomp(Xs_H3p3)$x %>% as_tibble %>% dplyr::select(PC1,PC2) %>% mutate(cluster=km_allH3p3$cluster) %>% ggplot(aes(PC1,PC2,colour=factor(cluster)))+geom_point(size=1.5,alpha=0.6)+coord_fixed()+theme_linedraw()+ggsci::scale_color_d3("category20")

file_path <- paste("./H3p3allcluster/", csvfilepath, "_kmeans__pcacluster_PC1PC2.pdf",sep="") 
ggsave(plot=pcacluster_save,file=file_path, width = 10, height = 6, dpi = 120)
print(pcacluster_save)

pcacluster_save <- prcomp(Xs_H3p3)$x %>% as_tibble %>% dplyr::select(PC1,PC3) %>% mutate(cluster=km_allH3p3$cluster) %>% ggplot(aes(PC1,PC3,colour=factor(cluster)))+geom_point(size=1.5,alpha=0.6)+coord_fixed()+theme_linedraw()+ggsci::scale_color_d3("category20")

file_path <- paste("./H3p3allcluster/", csvfilepath, "_kmeans__pcacluster_PC1PC3.pdf",sep="") 
ggsave(plot=pcacluster_save,file=file_path, width = 10, height = 6, dpi = 120)
print(pcacluster_save)


#================================================#
# mouseCTX 0438を参考に。

#------------------#
f_cluster <- function(x) x %>% group_by(group, type, time, cluster, seq) %>% summarise(avg=mean(val),se=sd(val)/sqrt(length(val))) %>% ungroup()
print(kmc_allH3p3_group %>% group_by(group, type, time) %>% summarise())

f_clusterp <- function(x) x %>% group_by(group, type, time, cluster, sep) %>% summarise(avg=mean(val),se=sd(val)/sqrt(length(val))) %>% ungroup()
print(kmc_allH3p3_group %>% group_by(group, type, time) %>% summarise()) #作図用

#-------#

cluster_save <- kmc_allH3p3_group %>%
ggplot(aes(time,val,group=type,colour=type))+ geom_abline(intercept=0,slope=0,linetype="dashed",colour="gray") +geom_line(aes(x=time,y=avg,colour=type),data=f_cluster)+geom_point()+facet_wrap(~cluster*seq,ncol=3)+ggsci::scale_color_npg()+theme_bw()+ theme(strip.text = element_text(size = 12),axis.text.x = element_text(vjust = 0.5), strip.background = element_blank(),  plot.title=element_text(size=5))  + ggtitle(gggglabel)+ggsci::scale_color_npg()  + ylab("z score")



file_path <- paste("./H3p3allcluster/", csvfilepath, "_cluster_type.pdf",sep="") 
ggsave(plot=cluster_save,file=file_path, width = 6, height = 6, dpi = 120)
#ggsave(plot=cluster_save,file=file_path, width = 6, height = 6, dpi = 120)
print(cluster_save)

#================================================#



```

```{r seq H3p3 cluster}
z_H3p3clus1 <- rrres_allH3p3 %>% filter(cluster=="1") %>% dplyr::rename(H3p3clus=cluster)
z_H3p3clus2 <- rrres_allH3p3 %>% filter(cluster=="2") %>% dplyr::rename(H3p3clus=cluster)
z_H3p3clus3 <- rrres_allH3p3 %>% filter(cluster=="3") %>% dplyr::rename(H3p3clus=cluster)
z_H3p3clus4 <- rrres_allH3p3 %>% filter(cluster=="4") %>% dplyr::rename(H3p3clus=cluster)
z_H3p3clus5 <- rrres_allH3p3 %>% filter(cluster=="5") %>% dplyr::rename(H3p3clus=cluster)
z_H3p3clus6 <- rrres_allH3p3 %>% filter(cluster=="6") %>% dplyr::rename(H3p3clus=cluster)

nrow(rrres_allH3p3)
nrow(z_H3p3clus1)
nrow(z_H3p3clus2)
nrow(z_H3p3clus3)
nrow(z_H3p3clus4)
nrow(z_H3p3clus5)
nrow(z_H3p3clus6)
nrow(rrres_allH3p3 %>% filter(is.na(cluster)))

rrres_allH3p3 %>% filter(ext_gene %in% c("Myh3","Ckm","Acta1","Tnnt2","Actb","Csrp3","Tpm2","Nsdhl","Myog"))
```



-----------------------------------------------

--------
## BRB result


```{r BRB result ext}

cluster_num <- 4 

##---- BRBのクラスタリング(LRT) の結果 -------#
#filepath_BRBcluster <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/LRT/DEG_fdr0p1__BRB0432lane2noumi_H3mm18_Dox_kmeans4__cluster_result.csv" #BRB等のDEGのリスト
#filepath_BRBallgene <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/H3mm18KO_3T3_Dox_normCount_genename.csv" #BRB等のDEGのリスト
#filepath_BRBzscore <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/LRT/clustering_XsLRTall__BRB0432lane2noumi_H3mm18_Dox.csv"
#filepath_BRBdef <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/deftable_BRB_noumi_new_190520_fin191205ver.txt"
#filepath_BRBdef <- "/home/guestA/o70578a/akuwakado/kuwakado/ChILSeq2/Komatsu_3T3_EGFP_H3mm18_Dox_chIl_0111NOVAseq/TSS_count/ChILAll_TSS_pm5kb_withATAC/deftable_BRB_noumi_new_190520_fin191205ver_20200625.txt"
#filepath_BRBChILATAC_def <- "/home/guestA/o70578a/akuwakado/kuwakado/ChILSeq2/Komatsu_3T3_EGFP_H3mm18_Dox_chIl_0111NOVAseq/TSS_count/ChILAll_TSS_pm5kb_withATAC/deftable_multicov_ChIL01100111_20200501_3T3_EGFP18_UI_DoxMinus_H3p3K27acK4Kme327me3_withATAC_withBRB.txt"
#filepath_BRBensemble <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/ensemble_list_asia.csv" #BRBの時のgene名リスト
#filepath_BRB_FC_all <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/LRT/all__BRB0432lane2noumi_H3mm18_Dox.csv" #BRB等のDEGのリスト
#filepath_BRB_2gun <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/2gun/BRB0432lane2noumi_H3mm18_Dox_results_fdr0p1__final191205.csv" #BRB等のDEGのリスト

filepath_BRBdef <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/deftable_BRB_noumi_new_190520_Last20200811ver.txt"

filepath_BRBcluster <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/LRT/DEG_fdr0p1__BRB0432lane2noumi_H3mm18_Dox_kmeans4__cluster_result.csv" #BRB等のDEGのリスト

filepath_BRBallgene <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/H3mm18KO_3T3_Dox_normCount_genename.csv" #BRB等のDEGのリスト
filepath_BRBzscore <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/H3mm18KO_3T3_Dox__zscore_type_all.csv"


##20200811 log2FC
filepath_BRB_FC_deseq <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/2gun/BRB0432lane2noumi_H3mm18_Dox_resultsall_fdr0p1__final191205_last200811.csv" #BRB等のDEGのリスト
filepath_BRB_normcount <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Last_Rserver_200811/BRB0432lane2noumi_H3mm18_Dox__normCount__final191205_last200811.csv"


#filepath_BRB_normcount <- "/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/BRB0432lane2noumi_H3mm18_Dox__normCount__final191205.csv"

#Final_Last_Rserver_200811

```


BRBのデータ (DEG,log2FCを抜き出す)

```{r DEG cluster prepare}
#--- DEGのリストの呼び出し --------#
cluster_BRBlist <- readr::read_csv(filepath_BRBcluster) %>% mutate(cluster=factor(cluster,c(1:cluster_num))) #BRB等のDEGのリスト

#--- log2 FCのリストの呼び出し --------#
re_BRB_all <- readr::read_csv(filepath_BRB_FC_deseq)

#--- BRBの全geneリストの呼び出し --------#　
all_BRBlist <- readr::read_csv(filepath_BRBallgene) #normcount
nrow(all_BRBlist)

#----#
zscore_BRB <- readr::read_csv(filepath_BRBzscore)
nrow(zscore_BRB)
#----#
def_BRB <- readr::read_tsv(filepath_BRBdef) %>% mutate(seq=factor(seq, c("BRB")))  %>% 
mutate(time1 = time, rep1=rep) %>% unite(time1,rep1,col="time_replicate")  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% mutate(type=factor(type,c("DoxPlus","DoxMinus"))) %>% mutate(rep=factor(rep, c("1", "2", "3", "4")))%>% mutate(time_replicate=factor(time_replicate,c("UI_1", "UI_2", "UI_3", "UI_4", "0h_1","0h_2","24h_1","24h_2","48h_1","48h_2","48h_3","48h_4")))

#----#
#FC_rank_all_BRBlist <- re_BRB_all %>% arrange(desc(abs(log2FoldChange))) %>% mutate(Rank=row_number()) #normcount
#nrow(FC_rank_all_BRBlist)
#print(FC_rank_all_BRBlist)

#%>% mutate(cluster=factor(cluster,c(1:cluster_num))) #BRB等のDEGのリスト


###--- DESeq2により計算したnormalized countのうち、BRB等のDEGのリストにあったものを書き出し ---###
#norm_table_select <- normalizedcount %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% full_join(cluster_BRBlist, ., by="ens_gene") #DEGリストのnormalized count
#readr::write_csv(norm_table_select, paste(".",folder_name, paste(RNAseq_cluster,"__normalizedcount__",folder_name,".csv",sep=""),sep="/")) #normalized count csvにgene名を付けた。
#------------------------------------------------------------------------------------------------# 
#----#
norm_BRB_def_original <- readr::read_csv(filepath_BRB_normcount)
#norm_BRB <- norm_BRB_def_original %>% dplyr::rename(Type=type,rep=replicate,norm=normalized) %>% mutate(type=case_when(Type=="Doxminus"~"DoxMinus",Type=="Doxplus"~"DoxPlus"))  %>% mutate(seq="BRB") %>% 
#mutate(time1 = time, rep1=rep) %>% unite(time1,rep1,col="time_replicate")  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% mutate(type=factor(type,c("DoxPlus","DoxMinus"))) %>% mutate(rep=factor(rep, c("1", "2", "3", "4")))%>% mutate(time_replicate=factor(time_replicate,c("UI_1", "UI_2", "UI_3", "UI_4", "0h_1","0h_2","24h_1","24h_2","48h_1","48h_2","48h_3","48h_4")))

norm_BRB <- norm_BRB_def_original %>% dplyr::rename(norm=normalized) %>% mutate(seq="BRB") %>% 
mutate(time1 = time, rep1=rep) %>% unite(time1,rep1,col="time_replicate")  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% mutate(type=factor(type,c("DoxPlus","DoxMinus"))) %>% mutate(rep=factor(rep, c("1", "2", "3", "4")))%>% mutate(time_replicate=factor(time_replicate,c("UI_1", "UI_2", "UI_3", "UI_4", "0h_1","0h_2","24h_1","24h_2","48h_1","48h_2","48h_3","48h_4")))


# %>% mutate(group=case_when(Group=="Doxminus_Diff0h"~"BRB_0h_DoxMinus",Group=="Doxplus_Diff0h"~"BRB_0h_DoxPlus"))

print(norm_BRB)
#----#

#BRB_2gun <- readr::read_csv(filepath_BRB_2gun)

#"/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/LRT/clustering_XsLRTall__BRB0432lane2noumi_H3mm18_Dox.csv"
#--- DEGのリストに位置情報を加える
#bedfile_cluster <- bedfile %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% dplyr::select(ens_gene,TSS_region,gene_region,Strand) %>% full_join(cluster_BRBlist, ., by="ens_gene") #位置情報あり
# dplyr::select(ens_gene,TSS_region,gene_region,Strand)

#NormCountBRBmat <- readr::read_csv("/home/guestA/o70578a/akuwakado/kuwakado/BRBSeq/H3mm18_Dox_0432lane2/Final_Rserver_191203/H3mm18KO_3T3_Dox_normCount_genename.csv")

```




## compare H3.3 cluster & BRB DEG cluster


```{r compare cluster & BRB}
list1 <- rrres_allH3p3 %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% group_by(cluster) %>% summarise(BRBcount=n(),genelist=paste(ext_gene,collapse=", "),IDlist=paste(ens_gene,collapse=", "))

#list2 <- rrres_H3p3clus2 %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% group_by(cluster) %>% summarise(BRBcount=n(),genelist=paste(ext_gene,collapse=", "),IDlist=paste(ens_gene,collapse=", "))

readr::write_csv(list1,"Compare_ChILATAC_cluster_vs_BRBDEGs.csv")
#readr::write_csv(list2,"Compare_ChILATAC_cluster2_vs_BRBDEGs.csv")

print(list1)

cluster1 <- cluster_BRBlist %>% filter(cluster=="1")
cluster2 <- cluster_BRBlist %>% filter(cluster=="2")
cluster3 <- cluster_BRBlist %>% filter(cluster=="3")
cluster4 <- cluster_BRBlist %>% filter(cluster=="4")


listclus <- rrres_allH3p3 %>% dplyr::select(ens_gene,ext_gene,cluster) %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% left_join(dplyr::select(cluster_BRBlist,ens_gene,cluster) %>% dplyr::rename(BRBclus=cluster)) %>% group_by(cluster,BRBclus) %>% summarise(count=n(),genelist=paste(ext_gene,collapse=", "),IDlist=paste(ens_gene,collapse=", "))

listclus2 <- listclus %>% dplyr::select(cluster,BRBclus,count) %>% mutate(BRBclus=paste("BRBcluster",BRBclus,sep="")) %>% spread(key=BRBclus,value=count, fill = 0)


readr::write_csv(listclus,"./Compare/Compare_ChILATAC_cluster_vs_BRBDEGs_summary.csv")
readr::write_csv(listclus2,"./Compare/Compare_ChILATAC_cluster_vs_BRBDEGs_summarycount.csv")

print(listclus)
print(listclus2)


```

```{r mosaic tile setup H3p3BRBlistclus, fig.width = 5, fig.height = 3}

#---#

H3p3BRBlistclus <- listclus2 %>%  mutate(H3p3cluster=paste(cluster,sep = ""))
#H3p3BRBlistclus <- listclus2 %>%  mutate(H3p3cluster=paste("H3p3cluster",cluster,sep = ""))

H3p3BRBlistclus_mat <- H3p3BRBlistclus %>% ungroup() %>% dplyr::select(-cluster,-H3p3cluster) %>% as.matrix()
rownames(H3p3BRBlistclus_mat) <- H3p3BRBlistclus$H3p3cluster

H3p3BRBlistclus_count <- H3p3BRBlistclus %>% ungroup() %>% dplyr::select(-cluster) %>% gather(key=sample,value=count,-H3p3cluster)

#---#

resultchisq <- chisq.test(H3p3BRBlistclus_mat)
resultchisq

#resultchisq$residuals
#resultchisq$expected
resultchisq$expected %>% sum()

#residuals <- resultchisq$residuals %>% as.data.frame(.) %>% tibble::rownames_to_column("H3p3cluster") %>% gather(key=sample,value=value,-(H3p3cluster)) %>% mutate(H3p3cluster=factor(H3p3cluster,c("H3p3cluster6","H3p3cluster5","H3p3cluster4","H3p3cluster3","H3p3cluster2","H3p3cluster1")))

#H3p3BRBlistclus_count_residuals <- residuals %>% left_join(H3p3BRBlistclus_count)  %>% mutate(H3p3cluster=factor(H3p3cluster,c("H3p3cluster6","H3p3cluster5","H3p3cluster4","H3p3cluster3","H3p3cluster2","H3p3cluster1")))

residuals <- resultchisq$residuals %>% as.data.frame(.) %>% tibble::rownames_to_column("H3p3cluster") %>% gather(key=sample,value=value,-(H3p3cluster)) %>% mutate(H3p3cluster=factor(H3p3cluster,c("1","2","3","4","5","6")))  %>% mutate(BRBcluster=gsub("BRBcluster","",sample)) %>% mutate(BRBcluster=factor(BRBcluster,c("4","3","2","1"))) 

H3p3BRBlistclus_count_residuals <- residuals %>% left_join(H3p3BRBlistclus_count)

#%>% mutate(H3p3cluster=factor(H3p3cluster,c("1","2","3","4","5","6")))

paste(resultchisq$method,"Residual",sep=": ")

chisq_plot <- H3p3BRBlistclus_count_residuals %>% ggplot(aes(x=H3p3cluster, y=BRBcluster, fill=value, label=count)) + geom_tile() + geom_text(aes(x=H3p3cluster, y=BRBcluster, label=as.character(count))) +  scale_fill_gradient2(low="blue", high="red", na.value="black", name="") + theme(axis.text.x  = element_text(angle = 90),title = element_text(size=2),legend.position = "top") + theme_minimal() + ylab("BRB DEG cluster") + xlab("H3.3 cluster")

chisq_plot
ggsave(file="./Compare/H3p3BRBlistclus.pdf", plot = chisq_plot, dpi = 100, width = 5, height = 3,limitsize = FALSE)


chisq_plot2 <- H3p3BRBlistclus_count_residuals %>% ggplot(aes(x=H3p3cluster, y=BRBcluster, fill=value, label=count)) + geom_tile() +  scale_fill_gradient2(low="blue", high="red", na.value="black", name="") + theme(axis.text.x  = element_text(angle = 90),title = element_text(size=2),legend.position = "top") + theme_minimal()  + ylab("BRB DEG cluster") + xlab("H3.3 cluster")

chisq_plot2
ggsave(file="./Compare/H3p3BRBlistclus_notitle.pdf", plot = chisq_plot2, dpi = 100, width = 5, height = 3,limitsize = FALSE)


H3p3BRBlistclus_count_residuals %>% readr::write_csv("./Compare/H3p3BRBlistclus.csv")

```



##--------- リストを保存 -------------#
#-- 確認 --#

rrres_allH3p3 %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% left_join(cluster_BRBlist %>% dplyr::rename(BRBclus=cluster)) %>% dplyr::select(cluster)%>% group_by(cluster) %>% summarise(count=n())

rrres_allH3p3 %>% filter(ens_gene %in% cluster_BRBlist$ens_gene) %>% left_join(cluster_BRBlist %>% dplyr::rename(BRBclus=cluster)) %>% dplyr::select(BRBclus,cluster) %>% group_by(BRBclus,cluster) %>% summarise(count=n())


##------------------------------------#


--------

# log2FCをまとめて計算する

#### H3p3 cluster3 のChIL normalized count

normalized count (BRB DEG) の　deftable等
（発現が低いものはCut）

まず、BRBで遺伝子発現が十分大きいものに絞り込む（Cut off リストの作成）

```{r norm count def table 200811ver}

Set_cutoff <- 10.0

## 各時刻の平均を計算し、normalized count > 10 を超えるものを抽出する。

#----- SKMとCTXのみ取り出す ---# 20191205
#norm_BRB_all <- norm_BRB %>% gather("sample", "normalized",-(ens_gene)) %>% inner_join(def, by = "sample")
#norm_BRB_all <- norm_BRB_all %>% filter(intact_CTX=="CTX"|intact_CTX=="SKM") %>% mutate(WT_KO=factor(WT_KO, c("H3mm18KO","WT"))) %>% mutate(Day=factor(Day, c("Day0","Day5","Day14"))) %>% mutate(intact_CTX=factor(intact_CTX, c("CTX","SKM")))

#notm_plotlist_cutoff <- norm_plotlist_all %>% annotate() %>% group_by(ens_gene, ext_gene, Day, intact_CTX) %>% summarize(groupMean=mean(normalized))  %>% ungroup() %>% dplyr::select(ens_gene, ext_gene) %>% unique()


norm_BRB_beforecutoff <- norm_BRB %>% group_by(ens_gene, ext_gene, seq, time) %>% summarize(groupMean=mean(norm))
nrow(norm_BRB_beforecutoff)
nrow(norm_BRB_beforecutoff %>% ungroup() %>% dplyr::select(ens_gene, ext_gene) %>% unique()) #この値をMAplotのx軸に使用

print("--- cut off ---")
norm_BRB_cutoff <- norm_BRB_beforecutoff %>% filter(groupMean > Set_cutoff) %>% ungroup()
nrow(norm_BRB_cutoff)

norm_BRB_cutoff_list <-norm_BRB_cutoff %>% dplyr::select(ens_gene, ext_gene) %>% unique()
nrow(norm_BRB_cutoff_list)

norm_BRB_beforecutoff %>% readr::write_csv("./log2FC/tables/Norm_BRB_groupMean.csv")
norm_BRB_cutoff  %>% readr::write_csv("log2FC/tables/Norm_BRB_groupMean_cutoff10.csv")
norm_BRB_cutoff_list  %>% readr::write_csv("log2FC/tables/Norm_BRB_groupMean_cutoff10_genelist.csv")

```

```{r extractRes3}

re_H3p3_FC_cutoff <- re_H3p3_all %>% mutate(seq="H3p3", time=gsub("group_H3p3_","",aspect)) %>% mutate(time=gsub("_Doxplus_vs_minus","",time)) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,  log2FoldChange,seq,time) %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene)

re_H3K4me3_FC_cutoff <- re_H3K4me3_all %>% mutate(seq="H3K4me3", time=gsub("group_H3K4me3_","",aspect)) %>% mutate(time=gsub("_Doxplus_vs_minus","",time)) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,  log2FoldChange,seq,time)  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene)

re_H3K27ac_FC_cutoff <- re_H3K27ac_all %>% mutate(seq="H3K27ac", time=gsub("group_H3K27ac_","",aspect)) %>% mutate(time=gsub("_Doxplus_vs_minus","",time)) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,  log2FoldChange,seq,time)  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene)

re_H3K27me3_FC_cutoff <- re_H3K27me3_all %>% mutate(seq="H3K27me3", time=gsub("group_H3K27me3_","",aspect)) %>% mutate(time=gsub("_Doxplus_vs_minus","",time)) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,  log2FoldChange,seq,time)  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene)

re_ATAC_FC_cutoff <- re_ATAC_all %>% mutate(seq="ATAC", time=gsub("group_ATAC_","",aspect)) %>% mutate(time=gsub("_Doxplus_vs_minus","",time)) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,  log2FoldChange,seq,time)  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene)

re_BRB_FC_cutoff <- re_BRB_all %>% mutate(seq="BRB", time=gsub("group_","",aspect)) %>% mutate(time=gsub("_Doxplus_vs_minus","",time)) %>% dplyr::select(ens_gene,ext_gene,biotype,chr,  log2FoldChange,seq,time)  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene)


filename <- "./log2FC/tables/log2FC_H3p3_cutoff10.csv"
print(filename)
readr::write_csv(re_H3p3_FC_cutoff,filename)
nrow(re_H3p3_FC_cutoff)

filename <- "./log2FC/tables/log2FC_H3K4me3_cutoff10.csv"
print(filename)
readr::write_csv(re_H3K4me3_FC_cutoff,filename)
nrow(re_H3K4me3_FC_cutoff)

filename <- "./log2FC/tables/log2FC_H3K27ac_cutoff10.csv"
print(filename)
readr::write_csv(re_H3K27ac_FC_cutoff,filename)
nrow(re_H3K27ac_FC_cutoff)

filename <- "./log2FC/tables/log2FC_H3K27me3_cutoff10.csv"
print(filename)
readr::write_csv(re_H3K27me3_FC_cutoff,filename)
nrow(re_H3K27me3_FC_cutoff)

filename <- "./log2FC/tables/lo2gFC_ATAC_cutoff10.csv"
print(filename)
readr::write_csv(re_ATAC_FC_cutoff,filename)
nrow(re_ATAC_FC_cutoff)

filename <- "./log2FC/tables/log2FC_BRB_cutoff10.csv"
print(filename)
readr::write_csv(re_BRB_FC_cutoff,filename)
nrow(re_BRB_FC_cutoff)


## 全て結合
re_all_FC_cutoff <- bind_rows(re_H3p3_FC_cutoff, re_H3K4me3_FC_cutoff) %>% bind_rows(re_H3K27ac_FC_cutoff) %>% bind_rows(re_H3K27me3_FC_cutoff) %>% bind_rows(re_ATAC_FC_cutoff) %>% bind_rows(re_BRB_FC_cutoff) %>% mutate(seq=factor(seq, c("H3p3", "H3K4me3","H3K27ac","H3K27me3","ATAC","BRB"))) %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) 

re_all_FC_cutoff

filename <- "./log2FC/tables/log2FC_ChILATACBRB_cutoff10.csv"
print(filename)
readr::write_csv(re_all_FC_cutoff,filename)
nrow(re_all_FC_cutoff)

```


```{r spread FC extractRes3}

spread_all_FC_cutoff <- re_all_FC_cutoff %>% group_by(ens_gene, ext_gene, biotype, chr, time) %>% spread(key=seq,value=log2FoldChange)
nrow(spread_all_FC_cutoff)
spread_all_FC_cutoff <- spread_all_FC_cutoff %>% left_join(dplyr::select(norm_BRB_cutoff, ens_gene, ext_gene, time, groupMean) %>% dplyr::rename(BRBgroupMean=groupMean))
nrow(spread_all_FC_cutoff)

filename <- "./log2FC/tables/Spread_log2FC_ChILATACBRB_cutoff10.csv"
print(filename)
readr::write_csv(spread_all_FC_cutoff,filename)


```

```{r spread FC extractRes3 add Cluster}

spread_all_FC_cutoff_clus <- spread_all_FC_cutoff %>% left_join(dplyr::select(rrres_allH3p3,ens_gene,cluster)) %>% dplyr::rename(H3p3cluster=cluster) %>% left_join(dplyr::select(cluster_BRBlist,ens_gene,cluster)) %>% dplyr::rename(BRBDEGcluster=cluster)

filename <- "./log2FC/tables/Spread_log2FC_ChILATACBRB_cutoff10__withCluster.csv"
print(filename)
readr::write_csv(spread_all_FC_cutoff_clus,filename)
nrow(spread_all_FC_cutoff_clus)


spread_all_FC_cutoff_clus %>% ungroup() %>% dplyr::select(ens_gene,H3p3cluster) %>% unique() %>% group_by(H3p3cluster) %>% summarize(H3p3_cutoff_count=n()) 
rrres_allH3p3  %>% group_by(cluster) %>% summarize(H3p3_geneAllcount=n())
```

```{r select spread FC Cluster H3p3 clus3}
f_gene_H3p3clus3 <- function(x) x %>% filter(H3p3cluster=="3")
f_gene_BRBclus3 <- function(x) x %>% filter(BRBDEGcluster=="3")
list_gene_qpcr <-  c("Acta1","Myh3","Ttn","Myog")

spread_all_FC_cutoff_H3p3clus3 <- spread_all_FC_cutoff_clus  %>% ungroup() %>% f_gene_H3p3clus3
nrow(spread_all_FC_cutoff_H3p3clus3)

filename <- "./log2FC/tables/Spread_log2FC_ChILATACBRB_cutoff10__withCluster__H3p3clus3.csv"
print(filename)
readr::write_csv(spread_all_FC_cutoff_H3p3clus3,filename)


spread_all_FC_cutoff_H3p3clus3
spread_all_FC_cutoff_H3p3clus3 %>% f_gene_H3p3clus3 %>% f_gene_BRBclus3
spread_all_FC_cutoff_H3p3clus3 %>% f_gene_H3p3clus3 %>% f_gene_BRBclus3 %>% filter(ext_gene %in% list_gene_qpcr)

H3p3clus3cutoff <- spread_all_FC_cutoff_H3p3clus3 %>% ungroup() %>% dplyr::select(ens_gene) %>% unique() %>% nrow() 
H3p3clus3cutoff_brbclus3 <- spread_all_FC_cutoff_H3p3clus3 %>% f_gene_BRBclus3 %>% ungroup() %>% dplyr::select(ens_gene) %>% unique() %>% nrow()

nrow(z_H3p3clus3)
H3p3clus3cutoff
H3p3clus3cutoff_brbclus3

```

  
```{r summary plot FC Cluster H3p3 clus3}

plot_all_FC_cutoff_H3p3clus3 <- spread_all_FC_cutoff_H3p3clus3 %>% dplyr::mutate(label_text = dplyr::case_when(ext_gene %in% list_gene_qpcr ~ ext_gene, TRUE ~ ""),shape = dplyr::case_when(ext_gene %in% list_gene_qpcr ~ "TRUE", TRUE ~ "FALSE"))

nrow(plot_all_FC_cutoff_H3p3clus3)

plot_all_FC_cutoff_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(!is.na(BRBgroupMean)) # groupMean > 10のみ残す

nrow(plot_all_FC_cutoff_H3p3clus3)


filename <- "./log2FC/tables/Plot_log2FC_ChILATACBRB_cutoff10__withCluster__H3p3clus3.csv"
print(filename)
readr::write_csv(plot_all_FC_cutoff_H3p3clus3,filename)

plot_all_FC_cutoff_H3p3clus3 %>% group_by(time) %>% summarise(count=n()) #図中の数
plot_all_FC_cutoff_H3p3clus3 %>% f_gene_BRBclus3 %>% group_by(time) %>% summarise(count=n()) #図中の数(BRB DEG cluster3のみ)

```





### Calculate Correlation H3.3 etc. vs BRB (All & BRB DEG Cluster3)

正規分布ならピアソンだが、今回正規分布ではないのでスピアマンの順位相関係数
を使う

```{r corre H3p3 cluster3 All UI-48h H3p3_BRB spearman}

Time_list <- c("UI","0h","24h","48h")


Count_FC_cutoff_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% ungroup() %>% group_by(time) %>% summarise(Plot_genes=n()) #図中の数


#######
print("~~ H3p3_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clusterAll: H3p3_BRB --"))
  corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3$H3p3, corr_H3p3clus3$BRB, method="spearman")
  #print(tttttt)
  
  if (i == 1) { 
      cortest_result_s <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".") %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("H3p3", "BRB",sep="_"))
  } 
  else {
      ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("H3p3", "BRB",sep="_"))
      cortest_result_s <- bind_rows(cortest_result_s, ssssss)
  }
}

print("~~ H3K4me3_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clusterAll: H3K4me3_BRB --"))
  corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3$H3K4me3, corr_H3p3clus3$BRB, method="spearman")
  #print(tttttt)
  
  ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("H3K4me3", "BRB",sep="_"))
  cortest_result_s <- bind_rows(cortest_result_s, ssssss)
}

print("~~ H3K27ac_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clusterAll: H3K27ac_BRB --"))
  corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3$H3K27ac, corr_H3p3clus3$BRB, method="spearman")
  #print(tttttt)
  

  ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("H3K27ac", "BRB",sep="_"))
  cortest_result_s <- bind_rows(cortest_result_s, ssssss)
}

print("~~ H3K27me3_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clusterAll: H3K27me3_BRB --"))
  corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3$H3K27me3, corr_H3p3clus3$BRB, method="spearman")
  #print(tttttt)
  

  ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("H3K27me3", "BRB",sep="_"))
  cortest_result_s <- bind_rows(cortest_result_s, ssssss)
}

print("~~ ATAC_BRB ~~")
for (i in 1:length(Time_list)) {

  if ((i == 1)|(i == 4)) { 
    print(paste("-----",Time_list[i], "--- H3p3clusterAll: ATAC_BRB --"))
    corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter((time==Time_list[i]))
    tttttt <- cor.test(corr_H3p3clus3$ATAC, corr_H3p3clus3$BRB, method="spearman")
    #print(tttttt)

    ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("ATAC", "BRB",sep="_"))
    cortest_result_s <- bind_rows(cortest_result_s, ssssss)
  } 
}

print("~~ H3p3_ATAC ~~")
for (i in 1:length(Time_list)) {

  if ((i == 1)|(i == 4)) { 
    print(paste("-----",Time_list[i], "--- H3p3clusterAll: H3p3_ATAC --"))
    corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter((time==Time_list[i]))
    tttttt <- cor.test(corr_H3p3clus3$H3p3, corr_H3p3clus3$ATAC, method="spearman")
    #print(tttttt)

    ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3All",Compare=paste("H3p3", "ATAC",sep="_"))
    cortest_result_s <- bind_rows(cortest_result_s, ssssss)
  } 
}


cortest_result_s_H3p3clus3All <- cortest_result_s %>% group_by(target,time,Compare) %>% spread(key="Cor_test",value=Value)  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% arrange(time)

cortest_result_s_H3p3clus3All <- cortest_result_s_H3p3clus3All %>% left_join(Count_FC_cutoff_H3p3clus3)
print(cortest_result_s_H3p3clus3All)

cortest_result_s_H3p3clus3All %>% readr::write_csv("./log2FC/tables/Cortest_result_spearman_H3p3clus3All.csv")


```





```{r corre H3p3 cluster3 BRB cluster3 UI-48h H3p3_BRB spearman}

Time_list <- c("UI","0h","24h","48h")
Count_FC_cutoff_H3p3clus3_BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% f_gene_BRBclus3 %>% ungroup() %>% group_by(time) %>% summarise(Plot_genes=n()) #図中の数(BRB DEG cluster3のみ)



#######
print("~~ H3p3_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clus3BRBclus3: H3p3_BRB --"))
  corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3BRBclus3$H3p3, corr_H3p3clus3BRBclus3$BRB, method="spearman")
  #print(tttttt)
  
  if (i == 1) { 
      cortest_result_s <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".") %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("H3p3", "BRB",sep="_"))
  } 
  else {
      ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("H3p3", "BRB",sep="_"))
      cortest_result_s <- bind_rows(cortest_result_s, ssssss)
  }
}

print("~~ H3K4me3_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clus3BRBclus3: H3K4me3_BRB --"))
  corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3BRBclus3$H3K4me3, corr_H3p3clus3BRBclus3$BRB, method="spearman")
  #print(tttttt)
  
  ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("H3K4me3", "BRB",sep="_"))
  cortest_result_s <- bind_rows(cortest_result_s, ssssss)
}

print("~~ H3K27ac_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clus3BRBclus3: H3K27ac_BRB --"))
  corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3BRBclus3$H3K27ac, corr_H3p3clus3BRBclus3$BRB, method="spearman")
  #print(tttttt)
  

  ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("H3K27ac", "BRB",sep="_"))
  cortest_result_s <- bind_rows(cortest_result_s, ssssss)
}

print("~~ H3K27me3_BRB ~~")
for (i in 1:length(Time_list)) {
  print(paste("-----",Time_list[i], "--- H3p3clus3BRBclus3: H3K27me3_BRB --"))
  corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time==Time_list[i]))
  tttttt <- cor.test(corr_H3p3clus3BRBclus3$H3K27me3, corr_H3p3clus3BRBclus3$BRB, method="spearman")
  #print(tttttt)
  

  ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("H3K27me3", "BRB",sep="_"))
  cortest_result_s <- bind_rows(cortest_result_s, ssssss)
}

print("~~ ATAC_BRB ~~")
for (i in 1:length(Time_list)) {

  if ((i == 1)|(i == 4)) { 
    print(paste("-----",Time_list[i], "--- H3p3clus3BRBclus3: ATAC_BRB --"))
    corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time==Time_list[i]))
    tttttt <- cor.test(corr_H3p3clus3BRBclus3$ATAC, corr_H3p3clus3BRBclus3$BRB, method="spearman")
    #print(tttttt)

    ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("ATAC", "BRB",sep="_"))
    cortest_result_s <- bind_rows(cortest_result_s, ssssss)
  } 
}

print("~~ H3p3_ATAC ~~")
for (i in 1:length(Time_list)) {

  if ((i == 1)|(i == 4)) { 
    print(paste("-----",Time_list[i], "--- H3p3clus3BRBclus3: H3p3_ATAC --"))
    corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time==Time_list[i]))
    tttttt <- cor.test(corr_H3p3clus3BRBclus3$H3p3, corr_H3p3clus3BRBclus3$ATAC, method="spearman")
    #print(tttttt)

    ssssss <- unlist(tttttt)  %>% as.data.frame() %>% tibble::rownames_to_column("Cor_test")  %>% as_tibble() %>% dplyr::rename(Value=".")  %>% mutate(time=Time_list[i],target="H3p3clus3BRBclus3",Compare=paste("H3p3", "ATAC",sep="_"))
    cortest_result_s <- bind_rows(cortest_result_s, ssssss)
  } 
}


cortest_result_s_H3p3clus3BRBclus3 <- cortest_result_s %>% group_by(target,time,Compare) %>% spread(key="Cor_test",value=Value)  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% arrange(time)

cortest_result_s_H3p3clus3BRBclus3 <- cortest_result_s_H3p3clus3BRBclus3 %>% left_join(Count_FC_cutoff_H3p3clus3_BRBclus3)
print(cortest_result_s_H3p3clus3BRBclus3)

cortest_result_s_H3p3clus3BRBclus3 %>% readr::write_csv("./log2FC/tables/Cortest_results_spearman_H3p3clus3BRBclus3.csv")


```


```{r rank, fig.width=5,fig.height=2}


# All
rank_corr_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% group_by(time) %>% mutate(rank_H3p3=rank(H3p3), rank_H3K4me3=rank(H3K4me3),  rank_H3K27ac=rank(H3K27ac), rank_H3K27me3=rank(H3K27me3), rank_ATAC=rank(ATAC), rank_BRB=rank(BRB))

#H3p3clus3BRBclus3
rank_corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3") %>% group_by(time) %>% mutate(rank_H3p3=rank(H3p3), rank_H3K4me3=rank(H3K4me3),  rank_H3K27ac=rank(H3K27ac), rank_H3K27me3=rank(H3K27me3), rank_ATAC=rank(ATAC), rank_BRB=rank(BRB))



###
fcplot <- rank_corr_H3p3clus3BRBclus3  %>% ggplot(aes(y=rank_BRB, x=rank_H3p3))  + facet_wrap(~time,nrow=1, scales = "free") + 
  xlim(0, NA) + ylim(0, NA) + geom_point(alpha = 0.6, size=1.0)+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank())

#+ xlim(0,60) + ylim(0,60)


fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3p3_BRB"),aes(x=0,y=60,label=paste(sprintf("%4.3e", estimate.rho),"  (",Plot_genes,"  genes)", sep="")), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 2.0)


#+ stat_smooth()
#+ stat_smooth(method = "lm", colour = "black", size = 1)
#+  geom_smooth(method = lm, se = FALSE)


fcplot


#rank_corr_H3p3clus3BRBclus3 <- plot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% filter((time=="48h")) %>% mutate(rank_H3p3=rank(H3p3), rank_H3K4me3=rank(H3K4me3),  rank_H3K27ac=rank(H3K27ac), rank_H3K27me3=rank(H3K27me3), rank_ATAC=rank(ATAC), rank_BRB=rank(BRB))

#cor.test(rank_corr_H3p3clus3BRBclus3$rank_H3p3, rank_corr_H3p3clus3BRBclus3$rank_BRB)


```

```{r extractRes3 norm}

### 並び替えのため

groups_BRB_arr <- c("BRB_UI_DoxMinus","BRB_UI_DoxPlus","BRB_0h_DoxMinus","BRB_0h_DoxPlus","BRB_24h_DoxMinus","BRB_24h_DoxPlus","BRB_48h_DoxMinus","BRB_48h_DoxPlus")
groups_ATAC_arr <- c("ATAC_UI_DoxMinus","ATAC_UI_DoxPlus","ATAC_48h_DoxMinus","ATAC_48h_DoxPlus")
groups_H3p3_arr <- c("H3p3_UI_DoxMinus","H3p3_UI_DoxPlus","H3p3_0h_DoxMinus","H3p3_0h_DoxPlus","H3p3_24h_DoxMinus","H3p3_24h_DoxPlus","H3p3_48h_DoxMinus","H3p3_48h_DoxPlus")
groups_H3K4me3_arr <- c("H3K4me3_UI_DoxMinus","H3K4me3_UI_DoxPlus","H3K4me3_0h_DoxMinus","H3K4me3_0h_DoxPlus","H3K4me3_24h_DoxMinus","H3K4me3_24h_DoxPlus","H3K4me3_48h_DoxMinus","H3K4me3_48h_DoxPlus")
groups_H3K27ac_arr <- c("H3K27ac_UI_DoxMinus","H3K27ac_UI_DoxPlus","H3K27ac_0h_DoxMinus","H3K27ac_0h_DoxPlus","H3K27ac_24h_DoxMinus","H3K27ac_24h_DoxPlus","H3K27ac_48h_DoxMinus","H3K27ac_48h_DoxPlus")
groups_H3K27me3_arr <- c("H3K27me3_UI_DoxMinus","H3K27me3_UI_DoxPlus","H3K27me3_0h_DoxMinus","H3K27me3_0h_DoxPlus","H3K27me3_24h_DoxMinus","H3K27me3_24h_DoxPlus","H3K27me3_48h_DoxMinus","H3K27me3_48h_DoxPlus")


groupt_BRB_arr <- c("BRB_DoxMinus","BRB_DoxPlus")
groupt_ATAC_arr <- c("ATAC_DoxMinus","ATAC_DoxPlus")
groupt_H3p3_arr <- c("H3p3_DoxMinus","H3p3_DoxPlus")
groupt_H3K4me3_arr <- c("H3K4me3_DoxMinus","H3K4me3_DoxPlus")
groupt_H3K27ac_arr <- c("H3K27ac_DoxMinus","H3K27ac_DoxPlus")
groupt_H3K27me3_arr <- c("H3K27me3_DoxMinus","H3K27me3_DoxPlus")


gggg_list2 <- c("ens_gene", "ext_gene", "time","BRBgroupMean", "H3p3cluster", "BRBDEGcluster", all_of(groupt_H3p3_arr),all_of(groupt_H3K4me3_arr),all_of(groupt_H3K27ac_arr),all_of(groupt_H3K27me3_arr),all_of(groupt_ATAC_arr),all_of(groupt_BRB_arr))

gggg_list1 <- c("ens_gene", "H3p3cluster", "BRBDEGcluster",all_of(groups_H3p3_arr),all_of(groups_H3K4me3_arr),all_of(groups_H3K27ac_arr),all_of(groups_H3K27me3_arr),all_of(groups_ATAC_arr),all_of(groups_BRB_arr))


## 全て結合


norm_chilatac_cutoff <- norm_plotlist_all  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene) %>% dplyr::select(ens_gene,sample,group,time,type,seq,rep,normalized)
norm_brb_cutoff <- norm_BRB  %>% filter(ens_gene %in% norm_BRB_cutoff$ens_gene) %>% dplyr::select(ens_gene,sample,group,time,type,seq,rep,norm) %>% rename(normalized=norm)

## 全て結合
norm_plotlist_cutoff <- bind_rows(norm_chilatac_cutoff,norm_brb_cutoff)  %>% mutate(seq=factor(seq, c("H3p3", "H3K4me3","H3K27ac","H3K27me3","ATAC","BRB"))) %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) # ChIL & ATAC & BRB (cut off 全て)

norm_typemean_cutoff <- norm_plotlist_cutoff  %>% group_by(ens_gene,group,time,type,seq) %>% summarise(TypeMean=mean(normalized))  %>% mutate(seq_type=paste(seq,type,sep="_"))

spread_norm_typemean_cutoff <- norm_typemean_cutoff %>% ungroup() %>% dplyr::select(ens_gene,time,seq_type,TypeMean)  %>% mutate(seq_type=factor(seq_type, c("H3p3_DoxMinus","H3p3_DoxPlus", "H3K4me3_DoxMinus","H3K4me3_DoxPlus","H3K27ac_DoxMinus","H3K27ac_DoxPlus","H3K27me3_DoxMinus","H3K27me3_DoxPlus","ATAC_DoxMinus","ATAC_DoxPlus","BRB_DoxMinus","BRB_DoxPlus"))) %>% arrange(seq_type) %>% spread(key=seq_type,value=TypeMean)

spread_norm_typemean_cutoff <- spread_norm_typemean_cutoff %>% left_join(dplyr::select(norm_BRB_cutoff, ens_gene, ext_gene, time, groupMean) %>% dplyr::rename(BRBgroupMean=groupMean))

nrow(spread_norm_typemean_cutoff)

spread_norm_typemean_cutoff_clus <- spread_norm_typemean_cutoff %>% left_join(dplyr::select(rrres_allH3p3,ens_gene,cluster)) %>% dplyr::rename(H3p3cluster=cluster) %>% left_join(dplyr::select(cluster_BRBlist,ens_gene,cluster)) %>% dplyr::rename(BRBDEGcluster=cluster)  %>% dplyr::select(all_of(gggg_list2))

nrow(spread_norm_typemean_cutoff_clus)

#####

time_norm_typemean_cutoff <- norm_typemean_cutoff %>% ungroup()  %>% dplyr::select(ens_gene,group,TypeMean) %>% group_by(ens_gene) %>% spread(key=group,value=TypeMean)

nrow(time_norm_typemean_cutoff)

time_norm_typemean_cutoff_clus <- time_norm_typemean_cutoff %>% left_join(dplyr::select(rrres_allH3p3,ens_gene,cluster)) %>% dplyr::rename(H3p3cluster=cluster) %>% left_join(dplyr::select(cluster_BRBlist,ens_gene,cluster)) %>% dplyr::rename(BRBDEGcluster=cluster) %>% dplyr::select(all_of(gggg_list1))


nrow(time_norm_typemean_cutoff_clus)

#####

filename <- "./Correlation/tables/normTypeMean_All_cutoff10.csv"
print(filename)
readr::write_csv(spread_norm_typemean_cutoff_clus,filename)
head(spread_norm_typemean_cutoff_clus)
nrow(spread_norm_typemean_cutoff_clus)





filename <- "./Correlation/tables/normTypeMean_All_cutoff10__timever.csv"
print(filename)
readr::write_csv(time_norm_typemean_cutoff_clus,filename)
head(time_norm_typemean_cutoff_clus)
nrow(time_norm_typemean_cutoff_clus)




```

```{r corr norm table}
#f_gene_H3p3clus3 <- function(x) x %>% filter(H3p3cluster=="3")
#f_gene_BRBclus3 <- function(x) x %>% filter(BRBDEGcluster=="3")
#list_gene_qpcr <-  c("Acta1","Myh3","Ttn","Myog")

corr_typemean_cutoff_H3p3clus3 <- time_norm_typemean_cutoff_clus   %>% ungroup() %>% f_gene_H3p3clus3
nrow(corr_typemean_cutoff_H3p3clus3)

corr_typemean_cutoff_H3p3clus3BRBclus3 <- time_norm_typemean_cutoff_clus   %>% ungroup() %>% f_gene_H3p3clus3 %>% f_gene_BRBclus3
nrow(corr_typemean_cutoff_H3p3clus3BRBclus3)



####

corr_typemean_cutoff_H3p3clus3_filter <- spread_norm_typemean_cutoff_clus  %>% filter(!is.na(BRBgroupMean))  %>% ungroup() %>% f_gene_H3p3clus3
nrow(corr_typemean_cutoff_H3p3clus3_filter)

corr_typemean_cutoff_H3p3clus3BRBclus3_filter <- spread_norm_typemean_cutoff_clus  %>% filter(!is.na(BRBgroupMean))  %>% ungroup() %>% f_gene_H3p3clus3 %>% f_gene_BRBclus3
nrow(corr_typemean_cutoff_H3p3clus3BRBclus3_filter)

corr_typemean_cutoff_H3p3clus3_filter %>% group_by(time) %>% summarise(count=n())
corr_typemean_cutoff_H3p3clus3BRBclus3_filter %>% group_by(time) %>% summarise(count=n())

#filename <- "./log2FC/tables/Spread_log2FC_ChILATACBRB_cutoff10__withCluster__H3p3clus3.csv"
#print(filename)
#readr::write_csv(spread_norm_typemean_cutoff_H3p3clus3,filename)


#spread_norm_typemean_cutoff_H3p3clus3
#spread_norm_typemean_cutoff_H3p3clus3 %>% f_gene_H3p3clus3 %>% f_gene_BRBclus3
#spread_norm_typemean_cutoff_H3p3clus3 %>% f_gene_H3p3clus3 %>% f_gene_BRBclus3 %>% filter(ext_gene %in% list_gene_qpcr)

#H3p3clus3cutoff <- spread_norm_typemean_cutoff_H3p3clus3 %>% ungroup() %>% dplyr::select(ens_gene) %>% unique() %>% nrow() 
#H3p3clus3cutoff_brbclus3 <- spread_norm_typemean_cutoff_H3p3clus3 %>% f_gene_BRBclus3 %>% ungroup() %>% dplyr::select(ens_gene) %>% unique() %>% nrow()

#nrow(z_H3p3clus3)
#H3p3clus3cutoff
#H3p3clus3cutoff_brbclus3

```

いずれかで BRB norm (group) > 10を満たすものでプロット

```{r norm corrplot norm typemean time, fig.width=10, fig.height=10}

library(corrplot)

#### calculate correlation

print("Genes list")
nrow(corr_typemean_cutoff_H3p3clus3)
nrow(corr_typemean_cutoff_H3p3clus3BRBclus3)

mydata.cor.All.alltime <- corr_typemean_cutoff_H3p3clus3 %>% ungroup() %>% dplyr::select(-"ens_gene", -"H3p3cluster", -"BRBDEGcluster") %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.alltime <- corr_typemean_cutoff_H3p3clus3BRBclus3 %>% ungroup() %>% dplyr::select(-"ens_gene", -"H3p3cluster", -"BRBDEGcluster") %>% cor(method = c("spearman"))


cor_All_alltime <- as.data.frame(mydata.cor.All.alltime) %>% tibble::rownames_to_column("group") %>% as_tibble 
cor_BRBclus3_alltime <- as.data.frame(mydata.cor.BRBclus3.alltime) %>% tibble::rownames_to_column("group") %>% as_tibble 

####


title_1 <- paste("H3.3 Cluster3 & BRB DEG Cluster3:",nrow(corr_typemean_cutoff_H3p3clus3BRBclus3),"genes",sep=" ")
title_2 <- paste("H3.3 Cluster3:",nrow(corr_typemean_cutoff_H3p3clus3),"genes",sep=" ")


breaksList = seq(-1, 1, by = 0.05)
Color__a0 <- rev(brewer.pal(n = 11, name = "RdYlBu"))
Color__a <- colorRampPalette(Color__a0)(length(breaksList))


gaps_1 <- c(8,16,24,32,36)



pheatmap::pheatmap(mydata.cor.BRBclus3.alltime, main = title_1, scale = "none", color = Color__a,breaks = breaksList, border_color=NA)
pheatmap::pheatmap(mydata.cor.All.alltime, main = title_2, scale = "none", color = Color__a,breaks = breaksList, border_color=NA)

pheatmap::pheatmap(mydata.cor.BRBclus3.alltime, main = title_1, scale = "none", color = Color__a,breaks = breaksList, border_color=NA, cluster_rows = FALSE, cluster_cols = FALSE, gaps_col=gaps_1, gaps_row=gaps_1)
pheatmap::pheatmap(mydata.cor.All.alltime, main = title_2, scale = "none", color = Color__a,breaks = breaksList, border_color=NA, cluster_rows = FALSE, cluster_cols = FALSE, gaps_col=gaps_1, gaps_row=gaps_1)


corrplot(mydata.cor.BRBclus3.alltime, diag = FALSE, col = Color__a)
corrplot(mydata.cor.All.alltime, diag = FALSE, col = Color__a)


#####

filename <- "./Correlation/tables/Cortest_normTypeMean_spearman_H3p3clus3All_cutoff10__timever.csv"
print(filename)
readr::write_csv(cor_All_alltime,filename)
print(cor_All_alltime)
nrow(cor_All_alltime)


filename <- "./Correlation/tables/Cortest_normTypeMean_spearman_H3p3clus3BRBclus3_cutoff10__timever.csv"
print(filename)
readr::write_csv(cor_BRBclus3_alltime,filename)
print(cor_BRBclus3_alltime)

#corrplot(mydata.cor.BRBclus3.alltime, diag = FALSE, type = "upper", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.alltime, diag = FALSE, type = "lower", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.alltime, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.alltime, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))


```

```{r norm corrplot norm typemean time-2, fig.width=10, fig.height=10}

## 表示リスト
gggg_plot_corlist <- c(all_of(groups_H3p3_arr),all_of(groups_H3K4me3_arr),all_of(groups_H3K27ac_arr),all_of(groups_BRB_arr))


cor_All_alltime_select <- cor_All_alltime %>% dplyr::select(group, all_of(gggg_plot_corlist)) %>% filter(group %in% gggg_plot_corlist)
mat_cor_All_alltime_select <- cor_All_alltime_select %>% dplyr::select(-group) %>% as.matrix()
rownames(mat_cor_All_alltime_select) <- cor_All_alltime_select$group


cor_BRBclus3_alltime_select <- cor_BRBclus3_alltime %>% dplyr::select(group, all_of(gggg_plot_corlist)) %>% filter(group %in% gggg_plot_corlist)
mat_cor_BRBclus3_alltime_select <- cor_BRBclus3_alltime_select %>% dplyr::select(-group) %>% as.matrix()
rownames(mat_cor_BRBclus3_alltime_select) <- cor_BRBclus3_alltime_select$group

gaps_2 <- c(8,16,24)

#### plot (select only)

#breaksList = seq(-1, 1, by = 0.05)
#Color__a0 <- rev(brewer.pal(n = 11, name = "RdYlBu"))
#Color__a <- colorRampPalette(Color__a0)(length(breaksList))

pheatmap::pheatmap(mat_cor_BRBclus3_alltime_select, main = title_1, scale = "none", color = Color__a,breaks = breaksList, border_color=NA)
pheatmap::pheatmap(mat_cor_All_alltime_select, main = title_2, scale = "none", color = Color__a,breaks = breaksList, border_color=NA)

pheatmap::pheatmap(mat_cor_BRBclus3_alltime_select, main = title_1, scale = "none", color = Color__a,breaks = breaksList, border_color=NA, cluster_rows = FALSE, cluster_cols = FALSE, gaps_col=gaps_2, gaps_row=gaps_2)
pheatmap::pheatmap(mat_cor_All_alltime_select, main = title_2, scale = "none", color = Color__a,breaks = breaksList, border_color=NA, cluster_rows = FALSE, cluster_cols = FALSE, gaps_col=gaps_2, gaps_row=gaps_2)


corrplot(mat_cor_BRBclus3_alltime_select, diag = FALSE, col = Color__a)
corrplot(mat_cor_All_alltime_select, diag = FALSE, col = Color__a)

```



  
```{r norm corrplot norm typemean time-3, fig.width=10, fig.height=10}

groups_UI_arr <- c(
  "H3p3_UI_DoxMinus","H3p3_UI_DoxPlus",
  "H3K4me3_UI_DoxMinus","H3K4me3_UI_DoxPlus",
  "H3K27ac_UI_DoxMinus","H3K27ac_UI_DoxPlus",
  "H3K27me3_UI_DoxMinus","H3K27me3_UI_DoxPlus",
  "ATAC_UI_DoxMinus","ATAC_UI_DoxPlus",
  "BRB_UI_DoxMinus","BRB_UI_DoxPlus")


groups_0h_arr <- c(
  "H3p3_0h_DoxMinus","H3p3_0h_DoxPlus",
  "H3K4me3_0h_DoxMinus","H3K4me3_0h_DoxPlus",
  "H3K27ac_0h_DoxMinus","H3K27ac_0h_DoxPlus",
  "H3K27me3_0h_DoxMinus","H3K27me3_0h_DoxPlus",
  "BRB_0h_DoxMinus","BRB_0h_DoxPlus")

groups_24h_arr <- c(
  "H3p3_24h_DoxMinus","H3p3_24h_DoxPlus",
  "H3K4me3_24h_DoxMinus","H3K4me3_24h_DoxPlus",
  "H3K27ac_24h_DoxMinus","H3K27ac_24h_DoxPlus",
  "H3K27me3_24h_DoxMinus","H3K27me3_24h_DoxPlus",
  "BRB_24h_DoxMinus","BRB_24h_DoxPlus")
  
groups_48h_arr <- c(
  "H3p3_48h_DoxMinus","H3p3_48h_DoxPlus",
  "H3K4me3_48h_DoxMinus","H3K4me3_48h_DoxPlus",
  "H3K27ac_48h_DoxMinus","H3K27ac_48h_DoxPlus",
  "H3K27me3_48h_DoxMinus","H3K27me3_48h_DoxPlus",
  "ATAC_48h_DoxMinus","ATAC_48h_DoxPlus",
  "BRB_48h_DoxMinus","BRB_48h_DoxPlus")


groups_UI_arr_s <- c(
  "H3p3_UI_DoxMinus","H3p3_UI_DoxPlus",
  "H3K4me3_UI_DoxMinus","H3K4me3_UI_DoxPlus",
  "H3K27ac_UI_DoxMinus","H3K27ac_UI_DoxPlus",
  "BRB_UI_DoxMinus","BRB_UI_DoxPlus")


groups_0h_arr_s <- c(
  "H3p3_0h_DoxMinus","H3p3_0h_DoxPlus",
  "H3K4me3_0h_DoxMinus","H3K4me3_0h_DoxPlus",
  "H3K27ac_0h_DoxMinus","H3K27ac_0h_DoxPlus",
  "BRB_0h_DoxMinus","BRB_0h_DoxPlus")

groups_24h_arr_s <- c(
  "H3p3_24h_DoxMinus","H3p3_24h_DoxPlus",
  "H3K4me3_24h_DoxMinus","H3K4me3_24h_DoxPlus",
  "H3K27ac_24h_DoxMinus","H3K27ac_24h_DoxPlus",
  "BRB_24h_DoxMinus","BRB_24h_DoxPlus")
  
groups_48h_arr_s <- c(
  "H3p3_48h_DoxMinus","H3p3_48h_DoxPlus",
  "H3K4me3_48h_DoxMinus","H3K4me3_48h_DoxPlus",
  "H3K27ac_48h_DoxMinus","H3K27ac_48h_DoxPlus",
  "BRB_48h_DoxMinus","BRB_48h_DoxPlus")


## 表示リスト
gggg_plot_corlist_time <- c(all_of(groups_UI_arr_s),all_of(groups_0h_arr_s),all_of(groups_24h_arr_s),all_of(groups_48h_arr_s))


cor_All_alltime_select2 <- cor_All_alltime %>% dplyr::select(group, all_of(gggg_plot_corlist_time)) %>% filter(group %in% gggg_plot_corlist_time) %>% mutate(group=factor(group,gggg_plot_corlist_time)) %>% arrange(group)
mat_cor_All_alltime_select2 <- cor_All_alltime_select2 %>% dplyr::select(-group) %>% as.matrix()
rownames(mat_cor_All_alltime_select2) <- cor_All_alltime_select2$group


cor_BRBclus3_alltime_select2 <- cor_BRBclus3_alltime%>% dplyr::select(group, all_of(gggg_plot_corlist_time)) %>% filter(group %in% gggg_plot_corlist_time) %>% mutate(group=factor(group,gggg_plot_corlist_time)) %>% arrange(group)
mat_cor_BRBclus3_alltime_select2 <- cor_BRBclus3_alltime_select2 %>% dplyr::select(-group) %>% as.matrix()
rownames(mat_cor_BRBclus3_alltime_select2) <- cor_BRBclus3_alltime_select2$group

gaps_2 <- c(8,16,24)

#### plot (select only)

pheatmap::pheatmap(mat_cor_BRBclus3_alltime_select2, main = title_1, scale = "none", color = Color__a,breaks = breaksList, border_color=NA, cluster_rows = FALSE, cluster_cols = FALSE, gaps_col=gaps_2, gaps_row=gaps_2)
pheatmap::pheatmap(mat_cor_All_alltime_select2, main = title_2, scale = "none", color = Color__a,breaks = breaksList, border_color=NA, cluster_rows = FALSE, cluster_cols = FALSE, gaps_col=gaps_2, gaps_row=gaps_2)



```  
  
  

```{r norm corrplot norm typemean all, fig.width=4,fig.height=4}

mydata.cor.BRBclus3.UI <- corr_typemean_cutoff_H3p3clus3BRBclus3_filter %>% ungroup() %>% filter(time=="UI") %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.0h <- corr_typemean_cutoff_H3p3clus3BRBclus3_filter %>% ungroup() %>% filter(time=="0h")   %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.24h <- corr_typemean_cutoff_H3p3clus3BRBclus3_filter %>% ungroup() %>% filter(time=="24h")  %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.48h <- corr_typemean_cutoff_H3p3clus3BRBclus3_filter %>% ungroup()  %>% filter(time=="48h")  %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))

mydata.cor.All.UI <- corr_typemean_cutoff_H3p3clus3_filter %>% ungroup() %>% filter(time=="UI")  %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))
mydata.cor.All.0h <- corr_typemean_cutoff_H3p3clus3_filter %>% ungroup() %>% filter(time=="0h")   %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))
mydata.cor.All.24h <- corr_typemean_cutoff_H3p3clus3_filter %>% ungroup() %>% filter(time=="24h")  %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))
mydata.cor.All.48h <- corr_typemean_cutoff_H3p3clus3_filter %>% ungroup()  %>% filter(time=="48h") %>% dplyr::select(BRB_DoxMinus, BRB_DoxPlus, H3p3_DoxMinus,H3p3_DoxPlus,H3K4me3_DoxMinus,H3K4me3_DoxPlus,H3K27ac_DoxMinus,H3K27ac_DoxPlus) %>% cor(method = c("spearman"))


#corrplot(mydata.cor.BRBclus3.UI, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.0h, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.24h, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.48h, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))

#corrplot(mydata.cor.All.UI, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.0h, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.24h, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.48h, diag = FALSE, method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))

corrplot(mydata.cor.BRBclus3.UI, diag = FALSE, method = "color")
corrplot(mydata.cor.BRBclus3.0h, diag = FALSE, method = "color")
corrplot(mydata.cor.BRBclus3.24h, diag = FALSE, method = "color")
corrplot(mydata.cor.BRBclus3.48h, diag = FALSE, method = "color")

corrplot(mydata.cor.All.UI, diag = FALSE, method = "color")
corrplot(mydata.cor.All.0h, diag = FALSE, method = "color")
corrplot(mydata.cor.All.24h, diag = FALSE, method = "color")
corrplot(mydata.cor.All.48h, diag = FALSE, method = "color")

corrplot(mydata.cor.BRBclus3.UI, diag = FALSE)
corrplot(mydata.cor.BRBclus3.0h, diag = FALSE)
corrplot(mydata.cor.BRBclus3.24h, diag = FALSE)
corrplot(mydata.cor.BRBclus3.48h, diag = FALSE)

corrplot(mydata.cor.All.UI, diag = FALSE)
corrplot(mydata.cor.All.0h, diag = FALSE)
corrplot(mydata.cor.All.24h, diag = FALSE)
corrplot(mydata.cor.All.48h, diag = FALSE)

#corrplot(mydata.cor.BRBclus3.UI, diag = FALSE, method = "color", col = cm.colors(100))
#corrplot(mydata.cor.BRBclus3.0h, diag = FALSE, method = "color", col = cm.colors(100))
#corrplot(mydata.cor.BRBclus3.24h, diag = FALSE, method = "color", col = cm.colors(100))
#corrplot(mydata.cor.BRBclus3.48h, diag = FALSE, method = "color", col = cm.colors(100))

#corrplot(mydata.cor.All.UI, diag = FALSE, method = "color", col = cm.colors(100))
#corrplot(mydata.cor.All.0h, diag = FALSE, method = "color", col = cm.colors(100))
#corrplot(mydata.cor.All.24h, diag = FALSE, method = "color", col = cm.colors(100))
#corrplot(mydata.cor.All.48h, diag = FALSE, method = "color", col = cm.colors(100))

#corrplot.mixed(cor(mydata.cor), order="hclust", tl.col="black")


#pheatmap::pheatmap(mydata.cor.All.UI,color=viridis::viridis(256),scale = "none")
#pheatmap::pheatmap(mydata.cor.All.0h,color=viridis::viridis(256),scale = "none")
#pheatmap::pheatmap(mydata.cor.All.24h,color=viridis::viridis(256),scale = "none")
#pheatmap::pheatmap(mydata.cor.All.48h,color=viridis::viridis(256),scale = "none")

#pheatmap::pheatmap(mydata.cor.BRBclus3.UI,color=viridis::viridis(256),scale = "none")
#pheatmap::pheatmap(mydata.cor.BRBclus3.0h,color=viridis::viridis(256),scale = "none")
#pheatmap::pheatmap(mydata.cor.BRBclus3.24h,color=viridis::viridis(256),scale = "none")
#pheatmap::pheatmap(mydata.cor.BRBclus3.48h,color=viridis::viridis(256),scale = "none")

#corrplot(mydata.cor.BRBclus3.UI, diag = FALSE, type = "upper", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.0h, diag = FALSE, type = "upper", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.24h, diag = FALSE, type = "upper", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.48h, diag = FALSE, type = "upper", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))

#corrplot(mydata.cor.All.UI, diag = FALSE, type = "lower", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.0h, diag = FALSE, type = "lower", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.24h, diag = FALSE, type = "lower", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.48h, diag = FALSE, type = "lower", method = "color", col = rev(brewer.pal(n = 10, name = "RdBu")))


#corrplot(mydata.cor.BRBclus3.UI, diag = FALSE, type = "upper", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.0h, diag = FALSE, type = "upper", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.24h, diag = FALSE, type = "upper", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.BRBclus3.48h, diag = FALSE, type = "upper", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))

#corrplot(mydata.cor.All.UI, diag = FALSE, type = "lower", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.0h, diag = FALSE, type = "lower", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.24h, diag = FALSE, type = "lower", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))
#corrplot(mydata.cor.All.48h, diag = FALSE, type = "lower", method = "square", col = rev(brewer.pal(n = 10, name = "RdBu")))

#corrplot.mixed(mydata.cor.BRBclus3.UI, lower.col = "black")
#corrplot.mixed(mydata.cor.BRBclus3.0h, lower.col = "black")
#corrplot.mixed(mydata.cor.BRBclus3.24h, lower.col = "black")
#corrplot.mixed(mydata.cor.BRBclus3.48h, lower.col = "black")

#corrplot(mydata.cor,palette = "PuOr")

```



```{r kokok}

library(corrplot)
# = read.csv("https://wiki.q-researchsoftware.com/images/b/b9/Ownership.csv", header = TRUE, fileEncoding="latin1")
#mydata.cor = cor(mydata)
#mydata.cor = cor(mydata, method = c("spearman")
#corrplot(mydata.cor)

#mydata.cor.UI <- rank_corr_H3p3clus3BRBclus3 %>% filter(time=="UI") %>% dplyr::select(ens_gene,ext_gene,time,H3p3,H3K4me3,H3K27ac,BRB) %>% tidyr::gather(key="seq",value=FC,-ens_gene,-ext_gene,-time) %>% mutate(time_seq=paste(time,seq,sep="_")) %>% ungroup %>% dplyr::select(ens_gene,time_seq,FC) %>% tidyr::spread(key = time_seq,value=FC) %>% dplyr::select(-ens_gene) %>% cor(method = c("spearman"))

#mydata.cor.0h <- rank_corr_H3p3clus3BRBclus3 %>% filter(time=="0h") %>% dplyr::select(ens_gene,ext_gene,time,H3p3,H3K4me3,H3K27ac,BRB) %>% tidyr::gather(key="seq",value=FC,-ens_gene,-ext_gene,-time) %>% mutate(time_seq=paste(time,seq,sep="_")) %>% ungroup %>% dplyr::select(ens_gene,time_seq,FC) %>% tidyr::spread(key = time_seq,value=FC) %>% dplyr::select(-ens_gene) %>% cor(method = c("spearman"))

#mydata.cor.24h <- rank_corr_H3p3clus3BRBclus3 %>% filter(time=="24h") %>% dplyr::select(ens_gene,ext_gene,time,H3p3,H3K4me3,H3K27ac,BRB) %>% tidyr::gather(key="seq",value=FC,-ens_gene,-ext_gene,-time) %>% mutate(time_seq=paste(time,seq,sep="_")) %>% ungroup %>% dplyr::select(ens_gene,time_seq,FC) %>% tidyr::spread(key = time_seq,value=FC) %>% dplyr::select(-ens_gene) %>% cor(method = c("spearman"))

#mydata.cor.48h <- rank_corr_H3p3clus3BRBclus3 %>% filter(time=="48h") %>% dplyr::select(ens_gene,ext_gene,time,H3p3,H3K4me3,H3K27ac,BRB) %>% tidyr::gather(key="seq",value=FC,-ens_gene,-ext_gene,-time) %>% mutate(time_seq=paste(time,seq,sep="_")) %>% ungroup %>% dplyr::select(ens_gene,time_seq,FC) %>% tidyr::spread(key = time_seq,value=FC) %>% dplyr::select(-ens_gene) %>% cor(method = c("spearman"))

mydata.cor.BRBclus3.UI <- rank_corr_H3p3clus3BRBclus3 %>% ungroup() %>% filter(time=="UI") %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.0h <- rank_corr_H3p3clus3BRBclus3 %>% ungroup() %>% filter(time=="0h")  %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.24h <- rank_corr_H3p3clus3BRBclus3 %>% ungroup() %>% filter(time=="24h") %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))
mydata.cor.BRBclus3.48h <- rank_corr_H3p3clus3BRBclus3 %>% ungroup()  %>% filter(time=="48h") %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))

mydata.cor.All.UI <- rank_corr_H3p3clus3 %>% ungroup() %>% filter(time=="UI") %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))
mydata.cor.All.0h <- rank_corr_H3p3clus3 %>% ungroup() %>% filter(time=="0h")  %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))
mydata.cor.All.24h <- rank_corr_H3p3clus3 %>% ungroup() %>% filter(time=="24h") %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))
mydata.cor.All.48h <- rank_corr_H3p3clus3 %>% ungroup()  %>% filter(time=="48h") %>% dplyr::select(BRB, H3p3,H3K4me3,H3K27ac) %>% cor(method = c("spearman"))


#corrplot.mixed(cor(mydata.cor), order="hclust", tl.col="black")
corrplot(mydata.cor.BRBclus3.UI, diag = FALSE, type = "upper", col = rev(brewer.pal(n = 10, name = "RdBu")))
corrplot(mydata.cor.BRBclus3.0h, diag = FALSE, type = "upper", col = rev(brewer.pal(n = 10, name = "RdBu")))
corrplot(mydata.cor.BRBclus3.24h, diag = FALSE, type = "upper", col = rev(brewer.pal(n = 10, name = "RdBu")))
corrplot(mydata.cor.BRBclus3.48h, diag = FALSE, type = "upper", col = rev(brewer.pal(n = 10, name = "RdBu")))

corrplot(mydata.cor.All.UI, diag = FALSE, type = "lower", col = rev(brewer.pal(n = 10, name = "RdBu")))
corrplot(mydata.cor.All.0h, diag = FALSE, type = "lower", col = rev(brewer.pal(n = 10, name = "RdBu")))
corrplot(mydata.cor.All.24h, diag = FALSE, type = "lower", col = rev(brewer.pal(n = 10, name = "RdBu")))
corrplot(mydata.cor.All.48h, diag = FALSE, type = "lower", col = rev(brewer.pal(n = 10, name = "RdBu")))

#corrplot.mixed(mydata.cor.BRBclus3.UI, lower.col = "black")
#corrplot.mixed(mydata.cor.BRBclus3.0h, lower.col = "black")
#corrplot.mixed(mydata.cor.BRBclus3.24h, lower.col = "black")
#corrplot.mixed(mydata.cor.BRBclus3.48h, lower.col = "black")

#corrplot(mydata.cor,palette = "PuOr")

```


spearmanとranking=>peason の相関係数が合っているかを調べること


```{r plot log2FC summary corrtest}

#%>% group_by(target,time,Compare)
#  group_by(aspect,gs_cat,gs_subcat) %>%
#  mutate(padj=p.adjust(pval,"BH")) %>% ungroup()
#Cortest_H3p3clus3All <- readr::read_csv("./log2FC/tables/Cortest_result_spearman_H3p3clus3All.csv") %>% mutate(text=paste(" All (",Plot_genes,") Cor: ",sprintf("%4.3e", estimate.cor),", p.val: ",sprintf("%4.3e", p.value),sep=""))   %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% arrange(time)
#Cortest_H3p3clus3BRBclus3 <- readr::read_csv("./log2FC/tables/Cortest_result_spearman_H3p3clus3BRBclus3.csv")  %>% mutate(text=paste("Clus3 (",Plot_genes,") Cor: ",sprintf("%4.3e", estimate.cor),", p.val: ",sprintf("%4.3e", p.value),sep=""))  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% arrange(time)


Cortest_H3p3clus3All <- readr::read_csv("./log2FC/tables/Cortest_result_spearman_H3p3clus3All.csv") %>% mutate(text=paste(" All (",Plot_genes,") Spearman Cor: ",sprintf("%4.3e", estimate.rho),", p.val: ",sprintf("%4.3e", p.value),sep=""))   %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% arrange(time)
Cortest_H3p3clus3BRBclus3 <- readr::read_csv("./log2FC/tables/Cortest_results_spearman_H3p3clus3BRBclus3.csv")  %>% mutate(text=paste("Clus3 (",Plot_genes,") Spearman Cor: ",sprintf("%4.3e", estimate.rho),", p.val: ",sprintf("%4.3e", p.value),sep=""))  %>% mutate(time=factor(time, c("UI", "0h","24h","48h"))) %>% arrange(time)

Cortest_H3p3clus3All %>% readr::write_csv("./log2FC/tables/Cortest_result_spearman_H3p3clus3All_forPlot.csv")
Cortest_H3p3clus3BRBclus3 %>% readr::write_csv("./log2FC/tables/Cortest_result_spearman_H3p3clus3BRBclus3_forPlot.csv")
```



```{r not shown log2FC}

Showplot_all_FC_cutoff_H3p3clus3 <- plot_all_FC_cutoff_H3p3clus3 %>% mutate(H3p3vsBRB=case_when(((abs(BRB)>2)|(abs(H3p3)>1.0)|is.na(H3p3))~"Not Shown",TRUE~"Shown"),H3K4me3vsBRB=case_when(((abs(BRB)>2)|(abs(H3K4me3)>1.5)|is.na(H3K4me3))~"Not Shown",TRUE~"Shown"),H3K27acvsBRB=case_when(((abs(BRB)>2)|(abs(H3K27ac)>1.0)|is.na(H3K27ac))~"Not Shown",TRUE~"Shown"),H3K27me3vsBRB=case_when(((abs(BRB)>2)|(abs(H3K27me3)>1.0)|is.na(H3K27me3))~"Not Shown",TRUE~"Shown"),ATACvsBRB=case_when(((abs(BRB)>2)|(abs(ATAC)>0.4)|is.na(ATAC))~"Not Shown",TRUE~"Shown"))

Show_H3p3vsBRB_H3p3clus3All <- Showplot_all_FC_cutoff_H3p3clus3 %>% ungroup %>% group_by(H3p3vsBRB,time) %>% summarize(count=n()) %>% rename(Plot=H3p3vsBRB) %>% mutate(Compare="H3p3_BRB")
Show_H3K4me3vsBRB_H3p3clus3All <- Showplot_all_FC_cutoff_H3p3clus3 %>% ungroup %>% group_by(H3K4me3vsBRB,time) %>% summarize(count=n()) %>% rename(Plot=H3K4me3vsBRB) %>% mutate(Compare="H3K4me3_BRB")
Show_H3K27acvsBRB_H3p3clus3All <- Showplot_all_FC_cutoff_H3p3clus3 %>% ungroup %>% group_by(H3K27acvsBRB,time) %>% summarize(count=n()) %>% rename(Plot=H3K27acvsBRB) %>% mutate(Compare="H3K27ac_BRB")
Show_H3K27me3vsBRB_H3p3clus3All <- Showplot_all_FC_cutoff_H3p3clus3 %>% ungroup %>% group_by(H3K27me3vsBRB,time) %>% summarize(count=n()) %>% rename(Plot=H3K27me3vsBRB)%>% mutate(Compare="H3K27me3_BRB")
Show_ATACvsBRB_H3p3clus3All <- Showplot_all_FC_cutoff_H3p3clus3 %>% ungroup %>% group_by(ATACvsBRB,time) %>% summarize(count=n()) %>% rename(Plot=ATACvsBRB)%>% mutate(Compare="ATAC_BRB")


Show__vsBRB_H3p3clus3All <- bind_rows(Show_H3p3vsBRB_H3p3clus3All,Show_H3K4me3vsBRB_H3p3clus3All) %>% bind_rows(Show_H3K27acvsBRB_H3p3clus3All) %>% bind_rows(Show_H3K27me3vsBRB_H3p3clus3All) %>% bind_rows(Show_ATACvsBRB_H3p3clus3All) %>% mutate(target="H3p3clus3All")  %>% mutate(Count=case_when(Plot=="Not Shown"~paste("(",count,")",sep=""),TRUE~as.character(count))) %>% mutate(Plot=factor(Plot, c("Shown","Not Shown"))) %>% arrange(Compare,time,Plot)


Show_H3p3vsBRB_H3p3clus3BRBclus3 <- Showplot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3") %>% ungroup %>% group_by(H3p3vsBRB,time) %>% summarize(count=n(),gene=paste(ext_gene,collapse = ",")) %>% rename(Plot=H3p3vsBRB) %>% mutate(Compare="H3p3_BRB")
Show_H3K4me3vsBRB_H3p3clus3BRBclus3 <- Showplot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% ungroup %>% group_by(H3K4me3vsBRB,time) %>% summarize(count=n(),gene=paste(ext_gene,collapse = ",")) %>% rename(Plot=H3K4me3vsBRB) %>% mutate(Compare="H3K4me3_BRB")
Show_H3K27acvsBRB_H3p3clus3BRBclus3 <- Showplot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% ungroup %>% group_by(H3K27acvsBRB,time) %>% summarize(count=n(),gene=paste(ext_gene,collapse = ",")) %>% rename(Plot=H3K27acvsBRB) %>% mutate(Compare="H3K27ac_BRB")
Show_H3K27me3vsBRB_H3p3clus3BRBclus3 <- Showplot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% ungroup %>% group_by(H3K27me3vsBRB,time) %>% summarize(count=n(),gene=paste(ext_gene,collapse = ",")) %>% rename(Plot=H3K27me3vsBRB) %>% mutate(Compare="H3K27me3_BRB")
Show_ATACvsBRB_H3p3clus3BRBclus3 <- Showplot_all_FC_cutoff_H3p3clus3 %>% filter(BRBDEGcluster=="3")%>% ungroup %>% group_by(ATACvsBRB,time) %>% summarize(count=n(),gene=paste(ext_gene,collapse = ",")) %>% rename(Plot=ATACvsBRB) %>% mutate(Compare="ATAC_BRB")

Show__vsBRB_H3p3clus3BRBclus3 <- bind_rows(Show_H3p3vsBRB_H3p3clus3BRBclus3,Show_H3K4me3vsBRB_H3p3clus3BRBclus3) %>% bind_rows(Show_H3K27acvsBRB_H3p3clus3BRBclus3) %>% bind_rows(Show_H3K27me3vsBRB_H3p3clus3BRBclus3) %>% bind_rows(Show_ATACvsBRB_H3p3clus3BRBclus3)  %>% mutate(target="H3p3clus3BRBclus3")  %>% mutate(Count=case_when(Plot=="Not Shown"~paste("(",count,")",sep=""),TRUE~as.character(count))) %>% mutate(Plot=factor(Plot, c("Shown","Not Shown"))) %>% arrange(Compare,time,Plot)


####
Show__vsBRB_H3p3clus3All
Show__vsBRB_H3p3clus3BRBclus3
Show__vsBRB_H3p3clus3BRBclus3 %>% filter(Plot=="Not Shown")

Show__vsBRB_H3p3clus3All %>% readr::write_csv("./log2FC/tables/PLotshow_H3p3clus3All.csv")
Show__vsBRB_H3p3clus3BRBclus3 %>% readr::write_csv("./log2FC/tables/PLotshow_H3p3clus3BRBclus3.csv")

summa__vsBRB_H3p3clus3All <- Show__vsBRB_H3p3clus3All %>% ungroup() %>% group_by(target, time, Compare) %>% summarize(Show=paste(Count,collapse = " "))
summa__vsBRB_H3p3clus3BRBclus3 <- Show__vsBRB_H3p3clus3BRBclus3 %>% ungroup() %>% group_by(target, time, Compare) %>% summarize(Show=paste(Count,collapse = " "))

summa__vsBRB_H3p3clus3All %>% readr::write_csv("./log2FC/tables/PLotshow_H3p3clus3All_summary.csv")
summa__vsBRB_H3p3clus3BRBclus3 %>% readr::write_csv("./log2FC/tables/PLotshow_H3p3clus3BRBclus3_summary.csv")


summa__vsBRB_H3p3clus3_ALL_BRBclus3 <- bind_rows(summa__vsBRB_H3p3clus3All,summa__vsBRB_H3p3clus3BRBclus3) %>% mutate(target=factor(target, c("H3p3clus3All","H3p3clus3BRBclus3"))) %>% arrange(Compare,time,target) %>% mutate(clus=gsub("H3p3clus3","",target)) %>% ungroup() %>% group_by(time,Compare) %>% summarize(show=paste(Show,collapse = " / "),Cluster=paste(clus,collapse = " / "))


summa__vsBRB_H3p3clus3_ALL_BRBclus3 %>% readr::write_csv("./log2FC/tables/PLotshow_H3p3clus3_All_and_BRBclus3_summary.csv")

summa__vsBRB_H3p3clus3_ALL_BRBclus3

```


```{r plot log2FC H3p3clus3 cut off dens, fig.width=4,fig.height=10}

#density_color_low <- "#ECE038"
density_color_low <- "#FFFFFF"
#density_color_high <- "#377EB8"
density_color_high <- "blue"
#density_color_low <- #FFFFFF"
#density_color_mid <- "yellow"
#density_color_high <- "red"

binsize <- 7


pppplottitle <- paste("log2 FC (Dox + vs -)\nBRB normalized count (Time, avg) > ",Set_cutoff,"\n H3.3 clus3: ",nrow(z_H3p3clus3)," genes\n Plot: ",H3p3clus3cutoff," genes\n BRB clus3:  ",H3p3clus3cutoff_brbclus3," genes",sep="")


###
fcplot <-plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3p3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3p3),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + xlab("H3.3") + ggtitle(pppplottitle)+ xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/log2FC_nolabel_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/log2FC_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3p3_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3p3_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3p3_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/log2FC_withcorr_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K4me3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3K4me3),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-1.5, 1.5) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/log2FC_nolabel_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/log2FC_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/log2FC_withcorr_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K27ac))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3K27ac),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/log2FC_nolabel_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/log2FC_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/log2FC_withcorr_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K27me3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3K27me3),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/log2FC_nolabel_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/log2FC_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/log2FC_withcorr_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3 %>% ggplot(aes(y=BRB, x=ATAC))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=ATAC),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-0.4, 0.4) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/log2FC_nolabel_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/log2FC_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="ATAC_BRB"),aes(x=-0.4,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="ATAC_BRB"),aes(x=-0.4,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="ATAC_BRB"),aes(x=-0.4,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/log2FC_withcorr_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
```





```{r plot log2FC H3p3clus3 cut off ec, fig.width=4,fig.height=10}

#density_color_low <- "#ECE038"
density_color_low <- "#FFFFFF"
#density_color_high <- "#377EB8"
density_color_high <- "blue"
#density_color_low <- #FFFFFF"
#density_color_mid <- "yellow"
#density_color_high <- "red"

binsize <- 7


pppplottitle <- paste("log2 FC (Dox + vs -)\nBRB normalized count (Time, avg) > ",Set_cutoff,"\n H3.3 clus3: ",nrow(z_H3p3clus3)," genes\n Plot: ",H3p3clus3cutoff," genes\n BRB clus3:  ",H3p3clus3cutoff_brbclus3," genes",sep="")


###
fcplot <-plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3p3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + stat_ellipse(type = "norm",color=density_color_high,size=0.2) + stat_ellipse(type = "norm",color="#000000",data=f_gene_BRBclus3,size=0.2) + geom_point(aes(y=BRB, x=H3p3),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + xlab("H3.3") + ggtitle(pppplottitle)+ xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_nolabel_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3p3_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3p3_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3p3_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_withcorr_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K4me3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + stat_ellipse(type = "norm",color=density_color_high,size=0.2) + stat_ellipse(type = "norm",color="#000000",data=f_gene_BRBclus3,size=0.2) + geom_point(aes(y=BRB, x=H3K4me3),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-1.5, 1.5) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_nolabel_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_withcorr_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K27ac))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + stat_ellipse(type = "norm",color=density_color_high,size=0.2) + stat_ellipse(type = "norm",color="#000000",data=f_gene_BRBclus3,size=0.2) + geom_point(aes(y=BRB, x=H3K27ac),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_nolabel_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_withcorr_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K27me3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + stat_ellipse(type = "norm",color=density_color_high,size=0.2) + stat_ellipse(type = "norm",color="#000000",data=f_gene_BRBclus3,size=0.2) + geom_point(aes(y=BRB, x=H3K27me3),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_nolabel_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_withcorr_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3 %>% ggplot(aes(y=BRB, x=ATAC))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + stat_ellipse(type = "norm",color=density_color_high,size=0.2) + stat_ellipse(type = "norm",color="#000000",data=f_gene_BRBclus3,size=0.2) + geom_point(aes(y=BRB, x=ATAC),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + ggtitle(pppplottitle)+ xlim(-0.4, 0.4) + ylim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_nolabel_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)


ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)


fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="ATAC_BRB"),aes(x=-0.4,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="ATAC_BRB"),aes(x=-0.4,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="ATAC_BRB"),aes(x=-0.4,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/ellipse/log2FC_ellipse_withcorr_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
```



For Check

```{r plot log2FC H3p3clus3 cut off dens Check, fig.width=4,fig.height=10}

#density_color_low <- "#ECE038"
density_color_low <- "#FFFFFF"
#density_color_high <- "#377EB8"
density_color_high <- "blue"
#density_color_low <- #FFFFFF"
#density_color_mid <- "yellow"
#density_color_high <- "red"

binsize <- 7


pppplottitle <- paste("log2 FC (Dox + vs -)\nBRB normalized count (Time, avg) > ",Set_cutoff,"\n H3.3 clus3: ",nrow(z_H3p3clus3)," genes\n Plot: ",H3p3clus3cutoff," genes\n BRB clus3:  ",H3p3clus3cutoff_brbclus3," genes",sep="")


###
fcplot <-plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3p3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3p3, color=BRBDEGcluster,shape=shape),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + scale_shape_manual(values=c(21, 19)) + xlab("H3.3") + ggtitle(pppplottitle)  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_nolimit_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

#fcplot
ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3p3_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3p3_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3p3_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_withcorr_H3p3clus3_cutoff__H3p3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K4me3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3K4me3, color=BRBDEGcluster,shape=shape),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + scale_shape_manual(values=c(21, 19))  + ggtitle(pppplottitle)  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_nolimit_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + xlim(-1.5, 1.5) + ylim(-2.0, 2.0)

#fcplot
ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)　+  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K4me3_BRB"),aes(x=-1.5,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_withcorr_H3p3clus3_cutoff__H3K4me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K27ac))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3K27ac, color=BRBDEGcluster,shape=shape),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank())  + scale_shape_manual(values=c(21, 19))  + ggtitle(pppplottitle) + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_nolimit_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

#fcplot
ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)　 +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K27ac_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_withcorr_H3p3clus3_cutoff__H3K27acvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(y=BRB, x=H3K27me3))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=H3K27me3, color=BRBDEGcluster,shape=shape),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank())  + scale_shape_manual(values=c(21, 19))  + ggtitle(pppplottitle)  + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_nolimit_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot   + xlim(-1.0, 1.0) + ylim(-2.0, 2.0)

#fcplot
ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8)　+  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="H3K27me3_BRB"),aes(x=-1.0,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_withcorr_H3p3clus3_cutoff__H3K27me3vsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
###
fcplot <- plot_all_FC_cutoff_H3p3clus3 %>% ggplot(aes(y=BRB, x=ATAC))  + facet_wrap(~time,ncol=1)  + stat_density2d(aes(fill=..density..), geom = "raster",contour = FALSE) + scale_fill_gradient(low = density_color_low, high = density_color_high)  + geom_abline(intercept=0,slope=0,colour="#000000",size=0.2) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density2d(aes(color=BRBDEGcluster),data=f_gene_BRBclus3,size=0.1,alpha = 0.5, bins=binsize) + geom_point(aes(y=BRB, x=ATAC, color=BRBDEGcluster,shape=shape),alpha = 0.6, size=1.0, data=f_gene_BRBclus3)  + scale_color_manual(values = c("#000000"))+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=10),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + scale_shape_manual(values=c(21, 19))  + ggtitle(pppplottitle) + geom_text_repel(aes(label = label_text), segment.color = "#000000",segment.size = 0.1) 

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_nolimit_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  + xlim(-0.4, 0.4) + ylim(-2.0, 2.0)

#fcplot
ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot <- fcplot  +  geom_text_repel(data=filter(Cortest_H3p3clus3BRBclus3,Compare=="ATAC_BRB"),aes(x=-0.4,y=1.8,label=text), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)  +  geom_text_repel(data=filter(Cortest_H3p3clus3All,Compare=="ATAC_BRB"),aes(x=-0.4,y=2.0,label=text), color = density_color_high, segment.color = density_color_high,segment.size = 0.1,size = 1.8) +  geom_text_repel(data=filter(summa__vsBRB_H3p3clus3_ALL_BRBclus3,Compare=="ATAC_BRB"),aes(x=-0.4,y=1.6,label=show), color = "#000000", segment.color = "#000000",segment.size = 0.1,size = 1.8)

ggsave(plot=fcplot,file="./log2FC/For_Check/log2FC_Check_withcorr_H3p3clus3_cutoff__ATACvsBRB.pdf", width = 3.5, height = 11, dpi = 360, limitsize = FALSE)

fcplot
```




20200817追加

```{r plot log2FC H3p3clus3 histgram dens, fig.width=4,fig.height=4}

#density_color_low <- "#ECE038"
density_color_low <- "#FFFFFF"
#density_color_high <- "#377EB8"
density_color_high <- "blue"
#density_color_low <- #FFFFFF"
#density_color_mid <- "yellow"
#density_color_high <- "red"

binsize <- 7


pppplottitle <- paste("log2 FC (Dox + vs -)\nBRB normalized count (Time, avg) > ",Set_cutoff,"\n H3.3 clus3: ",nrow(z_H3p3clus3)," genes\n Plot: ",H3p3clus3cutoff," genes\n BRB clus3:  ",H3p3clus3cutoff_brbclus3," genes",sep="")

###
fcplot <-plot_all_FC_cutoff_H3p3clus3 %>% ggplot(aes(x=BRB))  + facet_wrap(~time,ncol=1) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density(fill=density_color_high,color=density_color_high,alpha=0.5) + geom_density(fill="#000000",color="#000000",data=f_gene_BRBclus3,alpha=0.5) +theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=8),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank())+ xlim(-2.0, 2.0)

ggsave(plot=fcplot,file="./log2FC/histlog2FC__H3p3clus3_cutoff__BRB.pdf", width = 3.5, height = 3.5, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <-plot_all_FC_cutoff_H3p3clus3 %>% ggplot(aes(x=H3p3))  + facet_wrap(~time,ncol=1) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density(fill=density_color_high,color=density_color_high,alpha=0.5) + geom_density(fill="#000000",color="#000000",data=f_gene_BRBclus3,alpha=0.5) +theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=8),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + xlab("H3.3")+ xlim(-1.0, 1.0)

ggsave(plot=fcplot,file="./log2FC/histlog2FC__H3p3clus3_cutoff__H3p3.pdf", width = 3.5, height = 3.5, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3   %>% ggplot(aes(x=H3K4me3))  + facet_wrap(~time,ncol=1) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density(fill=density_color_high,color=density_color_high,alpha=0.5) + geom_density(fill="#000000",color="#000000",data=f_gene_BRBclus3,alpha=0.5) +theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=8),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank())+ xlim(-1.5, 1.5)

ggsave(plot=fcplot,file="./log2FC/histlog2FC__H3p3clus3_cutoff__H3K4me3.pdf", width = 3.5, height = 3.5, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3   %>% ggplot(aes(x=H3K27ac))  + facet_wrap(~time,ncol=1) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density(fill=density_color_high,color=density_color_high,alpha=0.5) + geom_density(fill="#000000",color="#000000",data=f_gene_BRBclus3,alpha=0.5) +theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=8),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + xlim(-1.0, 1.0)

ggsave(plot=fcplot,file="./log2FC/histlog2FC__H3p3clus3_cutoff__H3K27ac.pdf", width = 3.5, height = 3.5, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3   %>% ggplot(aes(x=H3K27me3))  + facet_wrap(~time,ncol=1) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density(fill=density_color_high,color=density_color_high,alpha=0.5) + geom_density(fill="#000000",color="#000000",data=f_gene_BRBclus3,alpha=0.5) +theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=8),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + xlim(-1.0, 1.0)

ggsave(plot=fcplot,file="./log2FC/histlog2FC__H3p3clus3_cutoff__H3K27me3.pdf", width = 3.5, height = 3.5, dpi = 360, limitsize = FALSE)

fcplot

###
fcplot <- plot_all_FC_cutoff_H3p3clus3  %>% ggplot(aes(x=ATAC))  + facet_wrap(~time,ncol=1) + geom_vline(xintercept = 0,colour="#000000",size=0.2) + geom_density(fill=density_color_high,color=density_color_high,alpha=0.5) + geom_density(fill="#000000",color="#000000",data=f_gene_BRBclus3,alpha=0.5)+theme_bw() + theme(axis.title = element_text(size=15),axis.text = element_text(size=8),axis.text.x = element_text(hjust = 0.5,vjust=1.0), legend.position = "right", strip.text=element_text(size=15),strip.background = element_blank(),title = element_text(size=8),panel.grid=element_blank()) + xlim(-0.4, 0.4)

ggsave(plot=fcplot,file="./log2FC/histlog2FC__H3p3clus3_cutoff__ATAC.pdf", width = 3.5, height = 3.5, dpi = 360, limitsize = FALSE)


fcplot
```


--------------------------------

#### クラスタリング (H3.3 cluster) の結果をGO

2020.4.21, 7.17修正 ver

```{r GO part1 Load list H3p3 cluster}
#20200421修正 ver
#20191206修正 ver

#z_heat_label_order_cluster6 <- z_heat_label_order_cluster %>% dplyr::select(ext_gene,heatmap_order,No,cluster_6) %>% mutate(heatmap_order=as.integer(heatmap_order),No=as.integer(No),cluster_6=as.integer(cluster_6))%>% arrange(heatmap_order) %>% left_join( dplyr::select(z_timedeg_s,ens_gene,ext_gene,biotype,chr))
#_____________#

## z_heat_label_order_cluster にクラスター番号が入っている

table_degcluster <- rrres_allH3p3 %>% filter(!is.na(cluster)) %>% arrange(cluster, ens_gene) %>% unique() %>% filter(!is.na(ens_gene))
degclusgene <- table_degcluster %>% group_by(cluster) %>% summarise(size=n()) %>% mutate(cluster=row_number())

table_degcluster <- table_degcluster %>% left_join(degclusgene %>% dplyr::select(cluster)) %>% arrange(cluster,ens_gene)

degclusgene
##### FDR setting ######
gofdr <- 0.1

#cluster_num <- 6
cluster_num <- nrow(degclusgene)

```

```{r go part2 clusterProfile H3p3 cluster}
# 20191206修正

library(clusterProfiler)
library(org.Mm.eg.db)

folder_path <- "./H3p3allcluster/clusterProfile/"

#-------------#
file_path <- paste(folder_path, "GO_newcluster_BPfdr0p1_generatio",sep="")
filename_csv <- file_path

file_path <- paste(folder_path, "GO_newcluster_BPfdr0p1_generatio_cluster",sep="")
filename_list <- file_path

print(filename_list)
print(filename_csv)

#例 filename_list <- "./LRT/clusterProfile/H3mm18KO_mouseCTX_BRB0438_day5_2gunfdr0p2_kmeans_BPfdr0p1_generatio_cluster"
#例 filename_csv <- "./LRT/clusterProfile/H3mm18KO_mouseCTX_BRB0438_day5_2gunfdr0p2_kemans_BPfdr0p1_generatio"
#-------------#

cluster_list <- as.list(NA) #初期化

for (i in 1:cluster_num) {
   pre_list <- as.list(NA)
   pre_list <- table_degcluster %>% filter(cluster==as.integer(i)) %>% dplyr::select(ens_gene) %>% as.list()
   names(pre_list) <- paste("ENSEMBL",as.character(i),sep="_")
 
   if (i == 1) { 
     cluster_list <- pre_list
   } 
   else cluster_list <- c(cluster_list, pre_list) 
}


for (i in 1:cluster_num) {
   print(paste(i, cluster_list[[i]] %>% tibble::enframe(name = NULL) %>% nrow(), sep=", "))
  
   pre_ego_BP <- enrichGO(gene = cluster_list[[i]],
                 OrgDb = "org.Mm.eg.db",
                 keyType = 'ENSEMBL',
                 ont = "BP",
                 pAdjustMethod = "BH",
                 pvalueCutoff  = gofdr, qvalueCutoff  = 1.0) 
   
   #20191211修正  pvalueCutoff  = fdr
   
   ## pvalue < qvalue < p.adjust ##
   # qvalueCutoff  = 0.3  qvalueCutoff  = 0.2 , qvalueCutoff  = 1.0

   #if (i == 1) { 
  #   table_ego_BP <- data.frame(pre_ego_BP) %>% mutate(cluster=as.integer(i))
  #   # リスト型からデータフレームへ変換
   #} 
   #else table_ego_BP <- table_ego_BP %>% bind_rows(data.frame(pre_ego_BP) %>% mutate(cluster=as.integer(i)))
                                                  
  if (i == 1) { 
     table_ego_BP <- data.frame(pre_ego_BP) %>% mutate(cluster=paste("cluster",as.character(i),sep=""))  # リスト型からデータフレームへ変換
   } 
   else table_ego_BP <- table_ego_BP %>% bind_rows(data.frame(pre_ego_BP) %>% mutate(cluster=paste("cluster",as.character(i),sep="")))
   
   #---- plot ---#
   BPplot <- dotplot(pre_ego_BP, showCategory=30, orderBy = "Count") #clusterProfile の機能で図を描く(191106修正) wrong orderBy parameter; set to default `orderBy = "x"`
   print(BPplot)
   ggsave(BPplot,file=paste(filename_list,as.character(i),".png",sep=""), width = 12, height = 12, dpi = 120)
   ggsave(BPplot,file=paste(filename_list,as.character(i),".pdf",sep=""), width = 12, height = 12, dpi = 120)
}

print(table_ego_BP %>% group_by(cluster) %>% summarize())

#------#
# データはtable_ego_BPに。

#------------------------------------------------------#
# テーブルを保存
# table_ego_BP_3t3_LRT2 <- table_ego_BP
#
table_ego_BP1 <- table_ego_BP %>% mutate(cluster=factor(cluster,c("cluster1","cluster2","cluster3","cluster4","cluster5","cluster6"))) %>% arrange(cluster,desc(Count)) #191106(200415)

#table_ego_BP1 <- table_ego_BP %>% arrange(cluster,desc(Count))  %>% left_join(dplyr::select(degclusgene, cluster)) #191106(200415)

readr::write_csv(table_ego_BP1,paste(filename_csv,".csv",sep=""))

```

```{r go part2-2 clusterProfile H3p3 cluster}

print(table_ego_BP %>% group_by(cluster) %>% summarize(cluster_3t3Dox_num = dplyr::n()))

# 先のテーブルのgeneIDをgene nameに置換する。(20191025)

tablego <- table_ego_BP1 %>% mutate(gene_name=geneID) %>% dplyr::select(-(qvalue))

for (i in 1:nrow(table_degcluster)) {
  tablego <- tablego %>% mutate(gene_name=gsub(gene_name, pattern=table_degcluster$ens_gene[i], replacement=table_degcluster$ext_gene[i], ignore.case = TRUE))
}

#print(tablego)

#readr::write_csv(tablego,paste(filename_csv,"_genename.csv",sep=""))

#------------------------------------------------------#


```


```{r GO save}

readr::write_csv(tablego,paste(filename_csv,"_genename.csv",sep=""))

```



---------------
```{r chromVAR}

library(chromVAR)
library(motifmatchr)
library(SummarizedExperiment)
library(Matrix)
library(ggplot2)
library(BiocParallel)
library(BSgenome.Mmusculus.UCSC.mm10)

list_plotallFCcutoff_H3p3clus3 <- dplyr::select(plot_all_FC_cutoff_H3p3clus3, ens_gene, ext_gene, biotype, chr, H3p3cluster, BRBDEGcluster) %>% unique()

TSSregion_H3p3clus3 <-  matome0_s %>% ungroup %>% dplyr::select(TSSstart,TSSend,ens_gene,score,strand,TSS,Start,End,position) %>% filter(ens_gene %in% list_plotallFCcutoff_H3p3clus3$ens_gene) %>% ungroup() %>% left_join(list_plotallFCcutoff_H3p3clus3)

#TSSPM10kb_H3p3clus3 <- TSSregion_H3p3clus3 %>% mutate(TSS_M10kb=TSS-10000,TSS_P10kb=TSS+10000) %>% dplyr::select(chr,TSS_M10kb,TSS_P10kb,ens_gene,score,strand,TSS,  Start,    End, position, ext_gene, biotype, H3p3cluster, BRBDEGcluster)

TSSPM10kb_H3p3clus3 <- TSSregion_H3p3clus3 %>% dplyr::select(chr,TSS,ens_gene, ext_gene,biotype, score, strand, H3p3cluster, BRBDEGcluster) %>% mutate(start=TSS-10000,end=TSS+10000) %>% dplyr::select(chr,start,end,ens_gene, score, strand,TSS, ext_gene,biotype,  H3p3cluster, BRBDEGcluster) %>% filter(BRBDEGcluster=="3")


TSSPM10kb_H3p3clus3_GR <- makeGRangesFromDataFrame(TSSPM10kb_H3p3clus3, keep.extra.columns = TRUE)

seqlevelsStyle(TSSPM10kb_H3p3clus3_GR) <- "UCSC"

#####

peakfile <- "./Motif/TSSPM10kb_H3p3clus3.csv"
print(peakfile)
readr::write_csv(TSSPM10kb_H3p3clus3,peakfile)
head(TSSPM10kb_H3p3clus3)
nrow(TSSPM10kb_H3p3clus3)


```


```{r read deftable}

def_bam_path <- "/home/guestA/o70578a/akuwakado/kuwakado/ChILSeq2/Komatsu_3T3_EGFP_H3mm18_Dox_chIl_0111NOVAseq/ChromVAR/ChromVAR_ChIL/H3K27ac_H3K27acpeak/deftable_ChromVAR_ChIL01100111_20200501_3T3_EGFP18_UI_DoxMinus_H3p3K27acK4Kme327me3.txt"

def_H3K27ac_bam <-  readr::read_tsv(file =def_bam_path) %>% filter(seq=="H3K27ac")


bamfiles <- def_H3K27ac_bam$file #bamfiles <- def_bam$peakcall_bam


fragment_counts <- getCounts(def_H3K27ac_bam$file, TSSPM10kb_H3p3clus3_GR, 
                              paired =  FALSE,  # ChILはペアではない。
                              by_rg = FALSE, 
                              colData = DataFrame(Cell_Type = def_H3K27ac_bam$group,sample_name = def_H3K27ac_bam$sample))


```

```{r fragment count}
length(fragment_counts)

print("---- fragment_counts ----")

slot(fragment_counts, "rowRanges") #fragment_counts @rowRanges

slot(fragment_counts, "colData")
slot(fragment_counts, "NAMES")
slot(fragment_counts, "metadata")
slot(fragment_counts, "elementMetadata")
slot(fragment_counts, "assays")


print("--------- save ----------------")

#-- save fragment count ----#
#--------#
type_depth <- slot(fragment_counts, "colData")  %>% as.data.frame() # 範囲
fffile <- sub(".csv","_typedepth.csv",peakfile)
print(fffile)
type_depth %>% readr::write_csv(fffile)
#--------#
f_c_range <- fragment_counts @rowRanges  %>% as.data.frame() # 範囲
fc_range <- f_c_range %>% mutate(seqnames1=seqnames,start1=start,end1=end) %>% unite(sten,c(start1,end1),sep="-") %>% unite(range,c(seqnames1,sten),sep=":")

f_c_count <- fragment_counts @assays @data$counts %>% as.matrix() %>% as.data.frame()  # カウント
#---#
f_c_range_count <- cbind(fc_range, f_c_count)
nrow(f_c_range_count)

fffile <- sub(".csv","_fragcounts.csv",peakfile)
print(fffile)
f_c_range_count %>% readr::write_csv(fffile)
#---------------------------#


```

```{r fragment count bias}

register(SerialParam())
fragment_counts_bias <- addGCBias(fragment_counts, genome = BSgenome.Mmusculus.UCSC.mm10) #ここでこけないように、Chrは確定されているものに設定。

#+++++++++++++++++++++++++++++++++++++++++++++++#
length(fragment_counts_bias)
print("---- fragment_counts_bias ----")
print("== rowRanges ==")
slot(fragment_counts_bias, "rowRanges")
print("== colData ==")
slot(fragment_counts_bias, "colData")
print("== NAMES ==")
slot(fragment_counts_bias, "NAMES")
print("== metadata ==")
slot(fragment_counts_bias, "metadata")
print("== elementMetadata ==")
slot(fragment_counts_bias, "elementMetadata")
print("== assays ==")
slot(fragment_counts_bias, "assays")

print("--------- save ----------------")

#-- save fragment count bias----#
f_c_range_bias <- fragment_counts_bias @rowRanges  %>% as.data.frame() # 範囲
fc_range_bias <- f_c_range_bias %>% mutate(seqnames1=seqnames,start1=start,end1=end) %>% unite(sten,c(start1,end1),sep="-") %>% unite(range,c(seqnames1,sten),sep=":")

f_c_count_bias <- fragment_counts_bias @assays @data$counts %>% as.matrix() %>% as.data.frame()  # カウント
#---#
f_c_range_count_bias <- cbind(fc_range_bias, f_c_count_bias)
nrow(f_c_range_count_bias)

fffile <- sub(".csv","_fragcounts_bias.csv",peakfile)
print(fffile)
f_c_range_count_bias %>% readr::write_csv(fffile)
#---------------------------#

```

```{r count filtered 1}
#counts_filtered <- filterSamples(fragment_counts_bias, min_depth = 1500, min_in_peaks = 0.15, shiny = FALSE)
counts_filtered_pre <- filterSamples(fragment_counts_bias, shiny = FALSE)

#++++++++++++++++++++++++++++++++++++++++++++++#
# If unspecified, min_in_peaks and min_depth cutoffs will be estimated based on data. min_in_peaks is set to 0.5 times the median proportion of fragments in peaks. min_depth is set to the maximum of 500 or 10 median library size.
#
# min_in_peaks: minimum fraction of samples within peaks
# min_depth:	minimum library size
# shiny:	make shiny gadget?
# ix_return:	return indices of sample to keep instead of subsetted counts object
#++++++++++++++++++++++++++++++++++++++++++++++#

length(counts_filtered_pre)
print("---- counts_filtered (pre) ----")
print("== rowRanges ==")
slot(counts_filtered_pre, "rowRanges")
print("== colData ==")
slot(counts_filtered_pre, "colData")
print("== NAMES ==")
slot(counts_filtered_pre, "NAMES")
print("== metadata ==")
slot(counts_filtered_pre, "metadata")
print("== elementMetadata ==")
slot(counts_filtered_pre, "elementMetadata")
print("== assays ==")
slot(counts_filtered_pre, "assays")
print("-------------------------")

```

```{r count filtered 2}
counts_filtered <- filterPeaks(counts_filtered_pre,non_overlapping=TRUE)

#++++++++++++++++++++++++++++++++++++++++++++++#
# if non_overlapping is set to true, when peaks overlap the overlapping peak with lower counts is removed
#
# min_fragments_per_peak:	minimum number of fragmints in peaks across all samples
# non_overlapping:	reduce peak set to non-overlapping peaks, see details
# ix_return:	return indices of peaks to keep instead of subsetted counts object
#++++++++++++++++++++++++++++++++++++++++++++++#

length(counts_filtered)
print("---- counts_filtered ----")
slot(counts_filtered, "rowRanges")
slot(counts_filtered, "colData")
slot(counts_filtered, "NAMES")
slot(counts_filtered, "metadata")
slot(counts_filtered, "elementMetadata")
slot(counts_filtered, "assays")
print("-------------------------")

#-- save counts_filtered ----#
f_c_range_countsfil <- counts_filtered @rowRanges  %>% as.data.frame() # 範囲
fc_range_countsfil <- f_c_range_countsfil %>% mutate(seqnames1=seqnames,start1=start,end1=end) %>% unite(sten,c(start1,end1),sep="-") %>% unite(range,c(seqnames1,sten),sep=":")

f_c_count_countsfil <- counts_filtered @assays @data$counts %>% as.matrix() %>% as.data.frame()  # カウント
#---#
f_c_range_count_countsfil <- cbind(fc_range_countsfil, f_c_count_countsfil)
nrow(f_c_range_count_countsfil)

fffile <- sub(".csv","_countsfilter.csv",peakfile)
print(fffile)
f_c_range_count_countsfil %>% readr::write_csv(fffile)
#---------------------------#

```

Raw deviations for background peaks
& Bias corrected deviations and Z-scores

```{r motif counts_filtered dev}

length(counts_filtered)

motifs <- chromVAR::getJasparMotifs(species = "Mus musculus", collection = "CORE") #OK
motif_ix <- matchMotifs(motifs, counts_filtered, genome = BSgenome.Mmusculus.UCSC.mm10)
#motifMatches(motif_ix) # Extract matches matrix from SummarizedExperiment result
dev <- computeDeviations(object = counts_filtered, annotations = motif_ix)


```

ChromVARで エラーのためここで終了(20200819)


#### differentialDev

(20200617追加)
https://greenleaflab.github.io/chromVAR/reference/differentialDeviations.html

data(mini_dev, package = "chromVAR")
difdev <- differentialDeviations(mini_dev, "Cell_Type")
differentialDeviations(object, groups, alternative = c("two.sided", "less","greater"), parametric = TRUE)
dev@colData

```{r differentialDev}
difdev <- differentialDeviations(dev, "Cell_Type", alternative = c("two.sided"))
#difdev <- differentialDeviations(dev, "Cell_Type", parametric = TRUE)
difdev_table <- difdev  %>% as_tibble(rownames = "motif_ID")  %>% left_join(motif_id)
difdev_table_select <- difdev_table %>% filter(p_value_adjusted<0.05) 

#-- save difdev_table with motifname ----#
fffile <- sub(".bed","_difdev_table_all.csv",peakfile)
print(fffile)
difdev_table %>% readr::write_csv(fffile)
nrow(difdev_table)
#-- save difdev_table with motifname ----#
fffile <- sub(".bed","_difdev_table_fdr0p01.csv",peakfile)
print(fffile)
difdev_table_select %>% readr::write_csv(fffile)
nrow(difdev_table_select)
#-----#

difdev_table_select %>% filter(motif_name %in% c("Myod1","Myog","Tcf12","Tcf21","Ascl2","FOS::JUN","Nfe2l2","Bach1::Mafk","RUNX1","Myb","Bcl6","Klf12","Klf4","Klf1","Gata4","Gata1","Rfx1","Spz1","Myc","Atoh1"))

#difdev_UI <- differentialDeviations(dev, Cell_Type, alternative = c("two.sided",  "Doxminus_UI_ATAC", "Doxplus_UI_ATAC"), parametric = TRUE)
#difdev_D48 <- differentialDeviations(dev, "Cell_Type", alternative = c("two.sided", "Doxminus_D48_ATAC", "Doxplus_D48_ATAC"), parametric = TRUE)


```
%>% filter(!motif_name %in% c("Myod1","Myog","Tcf12","Tcf21","Ascl2"))  %>% filter(!motif_name %in% c("FOS::JUN","Nfe2l2","Bach1::Mafk")) %>% filter(!motif_name %in% c("RUNX1","Myb"))  %>% filter(!motif_name %in% c("Bcl6","Klf12","Klf4","Klf1","Gata4","Gata1","Rfx1","Spz1","Myc","Atoh1"))

%>% filter(motif_name %in% c("Myod1","Myog","Tcf12","Tcf21","Ascl2"))  %>% filter(motif_name %in% c("FOS::JUN","Nfe2l2","Bach1::Mafk")) %>% filter(motif_name %in% c("RUNX1","Myb"))  %>% filter(motif_name %in% c("Bcl6","Klf12","Klf4","Klf1","Gata4","Gata1","Rfx1","Spz1","Myc","Atoh1"))



```{r save1 motif fragment_counts_bias dev}
#+++++++++++++++++++++++++++++++++++++++++++++++#

print("---- motif_ix ----------------")
motif_id <- motif_ix @colData %>% as_tibble(rownames = "motif_ID") %>% dplyr::rename(motif_name=name) #rename modif 20200616

print("-- motifMatches(motif_ix) --")
## motifMatches が . or | で 入っている
motifM_table <- motifMatches(motif_ix) %>% as.matrix()  %>% as.data.frame()  #motifM_table <- motifMatches(motif_ix) %>% as.matrix()  %>% as.data.frame() as_tibble()
#motifM_table %>% dplyr::select(1:3)
#colnames(motifM_table) #colnames(motifMatches(motif_ix))
ncol(motifM_table) #ncol(motifMatches(motif_ix))
motifM_table_range <- cbind(fc_range_countsfil, motifM_table)
ncol(motifM_table_range)
#motifM_table_range <- cbind(fc_range_bias %>% dplyr::select(range) ,motifM_table)

print("---- save motif ----")
#-- save motif id ----#
fffile <- sub(".bed","_motifID.csv",peakfile)
print(fffile)
motif_id %>% readr::write_csv(fffile)
#-- save motif match ----#
fffile <- sub(".bed","_motifMatches.csv",peakfile)
print(fffile)
motifM_table_range %>% readr::write_csv(fffile)


#==============================================#
#------ 複数の条件がある時に有用 ------#
print("========== dev ==========")
# slot(dev, "elementMetadata") #DataFrame with 128 rows and 3 columns
dev_table <- slot(dev, "elementMetadata") %>% as_tibble()  %>% dplyr::rename(motif_name=name) #rename modif 20200616
dev_table_withid <- dev_table %>% right_join(motif_id,., by = "motif_name") 
#dev_table_withid <- dev_table %>% right_join(motif_id,., by = "name")

print("-- dev (bias-corrected deviations) --")
dev_deviations <- dev @assays @data @listData$deviations %>% as_tibble(rownames = "motif_ID")
print("-- dev (z) --")
dev_z <- dev @assays @data @listData$z %>% as_tibble(rownames = "motif_ID")

print("---- save deviations ----")
#-- save deviations ----#
fffile <- sub(".bed","_deviations.csv",peakfile)
print(fffile)
dev_deviations %>% readr::write_csv(fffile)
#-- save dev (z) ----#
fffile <- sub(".bed","_dev_z.csv",peakfile)
print(fffile)
dev_z %>% readr::write_csv(fffile)

#==============================================#



```



